,url,repository_url,labels_url,comments_url,events_url,html_url,id,node_id,number,title,user,labels,state,locked,assignee,assignees,milestone,comments,created_at,updated_at,closed_at,author_association,active_lock_reason,body,reactions,timeline_url,performed_via_github_app,state_reason,draft,pull_request
0,https://api.github.com/repos/pandas-dev/pandas/issues/50427,https://api.github.com/repos/pandas-dev/pandas,https://api.github.com/repos/pandas-dev/pandas/issues/50427/labels{/name},https://api.github.com/repos/pandas-dev/pandas/issues/50427/comments,https://api.github.com/repos/pandas-dev/pandas/issues/50427/events,https://github.com/pandas-dev/pandas/issues/50427,1510092883,I_kwDOAA0YD85aAjBT,50427,BUG: `xticklables` is not correctly set when using `df.plot.box` with group-by ,"{'login': 'YMaster7', 'id': 73883983, 'node_id': 'MDQ6VXNlcjczODgzOTgz', 'avatar_url': 'https://avatars.githubusercontent.com/u/73883983?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/YMaster7', 'html_url': 'https://github.com/YMaster7', 'followers_url': 'https://api.github.com/users/YMaster7/followers', 'following_url': 'https://api.github.com/users/YMaster7/following{/other_user}', 'gists_url': 'https://api.github.com/users/YMaster7/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/YMaster7/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/YMaster7/subscriptions', 'organizations_url': 'https://api.github.com/users/YMaster7/orgs', 'repos_url': 'https://api.github.com/users/YMaster7/repos', 'events_url': 'https://api.github.com/users/YMaster7/events{/privacy}', 'received_events_url': 'https://api.github.com/users/YMaster7/received_events', 'type': 'User', 'site_admin': False}","[{'id': 76811, 'node_id': 'MDU6TGFiZWw3NjgxMQ==', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Bug', 'name': 'Bug', 'color': 'e10c02', 'default': False, 'description': None}, {'id': 1954720290, 'node_id': 'MDU6TGFiZWwxOTU0NzIwMjkw', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Needs%20Triage', 'name': 'Needs Triage', 'color': '0052cc', 'default': False, 'description': 'Issue that has not been reviewed by a pandas team member'}]",open,False,,[],,0,2022-12-24T13:21:13Z,2022-12-24T13:21:13Z,,NONE,,"### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the main branch of pandas.


### Reproducible Example

```python
import pandas as pd

df = pd.DataFrame({'value': [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
                   'label': ['c', 'c', 'c', 'c', 'b', 'b', 'b', 'b', 'a', 'a']})
df.label = pd.Categorical(df.label, categories=['c', 'b', 'a'], ordered=True)
df.plot.box(by='label')
```


### Issue Description

If I try to plot a dataframe and group it by a categorical column, the dataframe is not correctly plotted.

![image](https://user-images.githubusercontent.com/73883983/209437886-bdadce22-d9cd-4473-be5f-2ca9d5709b1b.png)


### Expected Behavior

Expected outcome should be like:

![image](https://user-images.githubusercontent.com/73883983/209437856-6d2287b7-1eac-4efd-b7ba-3c48cd03bb13.png)

This image was produced by the code below.

```
import pandas as pd

df = pd.DataFrame({'value': [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
                   'label': ['c', 'c', 'c', 'c', 'b', 'b', 'b', 'b', 'a', 'a']})
df.plot.box(by='label')
```

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 8dab54d6573f7186ff0c3b6364d5e4dd635ff3e7
python           : 3.9.15.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.15.79.1-microsoft-standard-WSL2
Version          : #1 SMP Wed Nov 23 01:01:46 UTC 2022
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : C.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.5.2
numpy            : 1.24.0
pytz             : 2022.7
dateutil         : 2.8.2
setuptools       : 65.6.3
pip              : 22.3.1
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.9.2
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : 8.7.0
pandas_datareader: None
bs4              : None
bottleneck       : None
brotli           : None
fastparquet      : None
fsspec           : None
gcsfs            : None
matplotlib       : 3.6.2
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : 1.9.3
snappy           : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None
tzdata           : None

</details>
","{'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50427/reactions', 'total_count': 0, '+1': 0, '-1': 0, 'laugh': 0, 'hooray': 0, 'confused': 0, 'heart': 0, 'rocket': 0, 'eyes': 0}",https://api.github.com/repos/pandas-dev/pandas/issues/50427/timeline,,,,
1,https://api.github.com/repos/pandas-dev/pandas/issues/50426,https://api.github.com/repos/pandas-dev/pandas,https://api.github.com/repos/pandas-dev/pandas/issues/50426/labels{/name},https://api.github.com/repos/pandas-dev/pandas/issues/50426/comments,https://api.github.com/repos/pandas-dev/pandas/issues/50426/events,https://github.com/pandas-dev/pandas/pull/50426,1510049222,PR_kwDOAA0YD85GKBO1,50426,ENH: Use cow for reindex_like,"{'login': 'phofl', 'id': 61934744, 'node_id': 'MDQ6VXNlcjYxOTM0NzQ0', 'avatar_url': 'https://avatars.githubusercontent.com/u/61934744?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/phofl', 'html_url': 'https://github.com/phofl', 'followers_url': 'https://api.github.com/users/phofl/followers', 'following_url': 'https://api.github.com/users/phofl/following{/other_user}', 'gists_url': 'https://api.github.com/users/phofl/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/phofl/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/phofl/subscriptions', 'organizations_url': 'https://api.github.com/users/phofl/orgs', 'repos_url': 'https://api.github.com/users/phofl/repos', 'events_url': 'https://api.github.com/users/phofl/events{/privacy}', 'received_events_url': 'https://api.github.com/users/phofl/received_events', 'type': 'User', 'site_admin': False}",[],open,False,,[],,0,2022-12-24T10:57:33Z,2022-12-24T10:57:33Z,,MEMBER,,"- [ ] xref #49473 (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.
","{'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50426/reactions', 'total_count': 0, '+1': 0, '-1': 0, 'laugh': 0, 'hooray': 0, 'confused': 0, 'heart': 0, 'rocket': 0, 'eyes': 0}",https://api.github.com/repos/pandas-dev/pandas/issues/50426/timeline,,,False,"{'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50426', 'html_url': 'https://github.com/pandas-dev/pandas/pull/50426', 'diff_url': 'https://github.com/pandas-dev/pandas/pull/50426.diff', 'patch_url': 'https://github.com/pandas-dev/pandas/pull/50426.patch', 'merged_at': None}"
2,https://api.github.com/repos/pandas-dev/pandas/issues/50425,https://api.github.com/repos/pandas-dev/pandas,https://api.github.com/repos/pandas-dev/pandas/issues/50425/labels{/name},https://api.github.com/repos/pandas-dev/pandas/issues/50425/comments,https://api.github.com/repos/pandas-dev/pandas/issues/50425/events,https://github.com/pandas-dev/pandas/pull/50425,1509890555,PR_kwDOAA0YD85GJgjx,50425,REF: Make Util functions accessible from C,"{'login': 'WillAyd', 'id': 609873, 'node_id': 'MDQ6VXNlcjYwOTg3Mw==', 'avatar_url': 'https://avatars.githubusercontent.com/u/609873?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/WillAyd', 'html_url': 'https://github.com/WillAyd', 'followers_url': 'https://api.github.com/users/WillAyd/followers', 'following_url': 'https://api.github.com/users/WillAyd/following{/other_user}', 'gists_url': 'https://api.github.com/users/WillAyd/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/WillAyd/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/WillAyd/subscriptions', 'organizations_url': 'https://api.github.com/users/WillAyd/orgs', 'repos_url': 'https://api.github.com/users/WillAyd/repos', 'events_url': 'https://api.github.com/users/WillAyd/events{/privacy}', 'received_events_url': 'https://api.github.com/users/WillAyd/received_events', 'type': 'User', 'site_admin': False}",[],open,False,,[],,1,2022-12-24T01:01:53Z,2022-12-24T01:08:32Z,,MEMBER,,"Came up when reviewing #50400 with @lithomas1 . Doesn't directly solve this, but a lot of our util.pyx functions are in essence C functions to begin with. Because they are implemented in Cython though, its a built difficult to use them in our non-Cython extensions like JSON.

This takes the code out of the Cython file and ports to a C header file. This is mostly a verbatim copy with some special considerations done for NA handling. Marking as draft because I'm not sure a header-only solution is better than a shared library, but need more time to mess around with the setup script. May also just wait until we implement meson fully","{'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50425/reactions', 'total_count': 0, '+1': 0, '-1': 0, 'laugh': 0, 'hooray': 0, 'confused': 0, 'heart': 0, 'rocket': 0, 'eyes': 0}",https://api.github.com/repos/pandas-dev/pandas/issues/50425/timeline,,,True,"{'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50425', 'html_url': 'https://github.com/pandas-dev/pandas/pull/50425', 'diff_url': 'https://github.com/pandas-dev/pandas/pull/50425.diff', 'patch_url': 'https://github.com/pandas-dev/pandas/pull/50425.patch', 'merged_at': None}"
3,https://api.github.com/repos/pandas-dev/pandas/issues/50424,https://api.github.com/repos/pandas-dev/pandas,https://api.github.com/repos/pandas-dev/pandas/issues/50424/labels{/name},https://api.github.com/repos/pandas-dev/pandas/issues/50424/comments,https://api.github.com/repos/pandas-dev/pandas/issues/50424/events,https://github.com/pandas-dev/pandas/pull/50424,1509807186,PR_kwDOAA0YD85GJOux,50424,PDEP6: ban upcasting in setitem-like operations,"{'login': 'MarcoGorelli', 'id': 33491632, 'node_id': 'MDQ6VXNlcjMzNDkxNjMy', 'avatar_url': 'https://avatars.githubusercontent.com/u/33491632?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/MarcoGorelli', 'html_url': 'https://github.com/MarcoGorelli', 'followers_url': 'https://api.github.com/users/MarcoGorelli/followers', 'following_url': 'https://api.github.com/users/MarcoGorelli/following{/other_user}', 'gists_url': 'https://api.github.com/users/MarcoGorelli/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/MarcoGorelli/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/MarcoGorelli/subscriptions', 'organizations_url': 'https://api.github.com/users/MarcoGorelli/orgs', 'repos_url': 'https://api.github.com/users/MarcoGorelli/repos', 'events_url': 'https://api.github.com/users/MarcoGorelli/events{/privacy}', 'received_events_url': 'https://api.github.com/users/MarcoGorelli/received_events', 'type': 'User', 'site_admin': False}","[{'id': 4253279386, 'node_id': 'LA_kwDOAA0YD879g-Sa', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/PDEP', 'name': 'PDEP', 'color': '4B02F8', 'default': False, 'description': 'pandas enhancement proposal'}]",open,False,,[],,2,2022-12-23T21:57:17Z,2022-12-24T13:07:39Z,,MEMBER,,"Original discussion: #39584

Turning into a PDEP as I think the `int -> float` warrants some care","{'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50424/reactions', 'total_count': 0, '+1': 0, '-1': 0, 'laugh': 0, 'hooray': 0, 'confused': 0, 'heart': 0, 'rocket': 0, 'eyes': 0}",https://api.github.com/repos/pandas-dev/pandas/issues/50424/timeline,,,False,"{'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50424', 'html_url': 'https://github.com/pandas-dev/pandas/pull/50424', 'diff_url': 'https://github.com/pandas-dev/pandas/pull/50424.diff', 'patch_url': 'https://github.com/pandas-dev/pandas/pull/50424.patch', 'merged_at': None}"
4,https://api.github.com/repos/pandas-dev/pandas/issues/50417,https://api.github.com/repos/pandas-dev/pandas,https://api.github.com/repos/pandas-dev/pandas/issues/50417/labels{/name},https://api.github.com/repos/pandas-dev/pandas/issues/50417/comments,https://api.github.com/repos/pandas-dev/pandas/issues/50417/events,https://github.com/pandas-dev/pandas/pull/50417,1509642413,PR_kwDOAA0YD85GIsqk,50417,TST: Add test for Period construction from str with zero nanos,"{'login': 'lithomas1', 'id': 47963215, 'node_id': 'MDQ6VXNlcjQ3OTYzMjE1', 'avatar_url': 'https://avatars.githubusercontent.com/u/47963215?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/lithomas1', 'html_url': 'https://github.com/lithomas1', 'followers_url': 'https://api.github.com/users/lithomas1/followers', 'following_url': 'https://api.github.com/users/lithomas1/following{/other_user}', 'gists_url': 'https://api.github.com/users/lithomas1/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/lithomas1/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/lithomas1/subscriptions', 'organizations_url': 'https://api.github.com/users/lithomas1/orgs', 'repos_url': 'https://api.github.com/users/lithomas1/repos', 'events_url': 'https://api.github.com/users/lithomas1/events{/privacy}', 'received_events_url': 'https://api.github.com/users/lithomas1/received_events', 'type': 'User', 'site_admin': False}","[{'id': 127685, 'node_id': 'MDU6TGFiZWwxMjc2ODU=', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Testing', 'name': 'Testing', 'color': 'C4A000', 'default': False, 'description': 'pandas testing functions or related to the test suite'}, {'id': 60635328, 'node_id': 'MDU6TGFiZWw2MDYzNTMyOA==', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Period', 'name': 'Period', 'color': 'eb6420', 'default': False, 'description': 'Period data type'}]",open,False,,[],,2,2022-12-23T18:35:05Z,2022-12-23T22:23:39Z,,MEMBER,,"- [ ] xref #50149 (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.
","{'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50417/reactions', 'total_count': 0, '+1': 0, '-1': 0, 'laugh': 0, 'hooray': 0, 'confused': 0, 'heart': 0, 'rocket': 0, 'eyes': 0}",https://api.github.com/repos/pandas-dev/pandas/issues/50417/timeline,,,False,"{'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50417', 'html_url': 'https://github.com/pandas-dev/pandas/pull/50417', 'diff_url': 'https://github.com/pandas-dev/pandas/pull/50417.diff', 'patch_url': 'https://github.com/pandas-dev/pandas/pull/50417.patch', 'merged_at': None}"
5,https://api.github.com/repos/pandas-dev/pandas/issues/50416,https://api.github.com/repos/pandas-dev/pandas,https://api.github.com/repos/pandas-dev/pandas/issues/50416/labels{/name},https://api.github.com/repos/pandas-dev/pandas/issues/50416/comments,https://api.github.com/repos/pandas-dev/pandas/issues/50416/events,https://github.com/pandas-dev/pandas/pull/50416,1509630900,PR_kwDOAA0YD85GIqBN,50416,CLN: Use exclusions in determination of _selected_obj,"{'login': 'rhshadrach', 'id': 45562402, 'node_id': 'MDQ6VXNlcjQ1NTYyNDAy', 'avatar_url': 'https://avatars.githubusercontent.com/u/45562402?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/rhshadrach', 'html_url': 'https://github.com/rhshadrach', 'followers_url': 'https://api.github.com/users/rhshadrach/followers', 'following_url': 'https://api.github.com/users/rhshadrach/following{/other_user}', 'gists_url': 'https://api.github.com/users/rhshadrach/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/rhshadrach/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/rhshadrach/subscriptions', 'organizations_url': 'https://api.github.com/users/rhshadrach/orgs', 'repos_url': 'https://api.github.com/users/rhshadrach/repos', 'events_url': 'https://api.github.com/users/rhshadrach/events{/privacy}', 'received_events_url': 'https://api.github.com/users/rhshadrach/received_events', 'type': 'User', 'site_admin': False}","[{'id': 233160, 'node_id': 'MDU6TGFiZWwyMzMxNjA=', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Groupby', 'name': 'Groupby', 'color': '729FCF', 'default': False, 'description': None}, {'id': 211029535, 'node_id': 'MDU6TGFiZWwyMTEwMjk1MzU=', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Clean', 'name': 'Clean', 'color': '207de5', 'default': False, 'description': None}]",open,False,,[],"{'url': 'https://api.github.com/repos/pandas-dev/pandas/milestones/42', 'html_url': 'https://github.com/pandas-dev/pandas/milestone/42', 'labels_url': 'https://api.github.com/repos/pandas-dev/pandas/milestones/42/labels', 'id': 1945020, 'node_id': 'MDk6TWlsZXN0b25lMTk0NTAyMA==', 'number': 42, 'title': '2.0', 'description': '', 'creator': {'login': 'jreback', 'id': 953992, 'node_id': 'MDQ6VXNlcjk1Mzk5Mg==', 'avatar_url': 'https://avatars.githubusercontent.com/u/953992?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/jreback', 'html_url': 'https://github.com/jreback', 'followers_url': 'https://api.github.com/users/jreback/followers', 'following_url': 'https://api.github.com/users/jreback/following{/other_user}', 'gists_url': 'https://api.github.com/users/jreback/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/jreback/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/jreback/subscriptions', 'organizations_url': 'https://api.github.com/users/jreback/orgs', 'repos_url': 'https://api.github.com/users/jreback/repos', 'events_url': 'https://api.github.com/users/jreback/events{/privacy}', 'received_events_url': 'https://api.github.com/users/jreback/received_events', 'type': 'User', 'site_admin': False}, 'open_issues': 53, 'closed_issues': 851, 'state': 'open', 'created_at': '2016-08-16T13:21:24Z', 'updated_at': '2022-12-24T10:01:12Z', 'due_on': '2023-01-31T08:00:00Z', 'closed_at': None}",2,2022-12-23T18:26:30Z,2022-12-23T21:25:49Z,,MEMBER,,"- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Toward #46944.

Currently we use whether a grouping is in-axis or not to determine if it should make it into `_selected_obj`. This is almost the same as using `exclusions`, except for a TimeGrouper. Here, the grouping data may arise from a column and hence should be excluded, however the grouping data is not the column itself but rather derived from the column, so it is not technically in-axis. Such columns should not make it into `_selected_obj`, so we should prefer using excluded directly rather than in-axis.

This does not change the behavior of Resampler because Resampler is a subclass of BaseGroupBy which does not have the selection context.","{'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50416/reactions', 'total_count': 0, '+1': 0, '-1': 0, 'laugh': 0, 'hooray': 0, 'confused': 0, 'heart': 0, 'rocket': 0, 'eyes': 0}",https://api.github.com/repos/pandas-dev/pandas/issues/50416/timeline,,,False,"{'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50416', 'html_url': 'https://github.com/pandas-dev/pandas/pull/50416', 'diff_url': 'https://github.com/pandas-dev/pandas/pull/50416.diff', 'patch_url': 'https://github.com/pandas-dev/pandas/pull/50416.patch', 'merged_at': None}"
6,https://api.github.com/repos/pandas-dev/pandas/issues/50415,https://api.github.com/repos/pandas-dev/pandas,https://api.github.com/repos/pandas-dev/pandas/issues/50415/labels{/name},https://api.github.com/repos/pandas-dev/pandas/issues/50415/comments,https://api.github.com/repos/pandas-dev/pandas/issues/50415/events,https://github.com/pandas-dev/pandas/pull/50415,1509593635,PR_kwDOAA0YD85GIhv8,50415,Add cow mechanism to rename_axis,"{'login': 'phofl', 'id': 61934744, 'node_id': 'MDQ6VXNlcjYxOTM0NzQ0', 'avatar_url': 'https://avatars.githubusercontent.com/u/61934744?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/phofl', 'html_url': 'https://github.com/phofl', 'followers_url': 'https://api.github.com/users/phofl/followers', 'following_url': 'https://api.github.com/users/phofl/following{/other_user}', 'gists_url': 'https://api.github.com/users/phofl/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/phofl/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/phofl/subscriptions', 'organizations_url': 'https://api.github.com/users/phofl/orgs', 'repos_url': 'https://api.github.com/users/phofl/repos', 'events_url': 'https://api.github.com/users/phofl/events{/privacy}', 'received_events_url': 'https://api.github.com/users/phofl/received_events', 'type': 'User', 'site_admin': False}",[],open,False,,[],,0,2022-12-23T17:43:57Z,2022-12-24T10:41:23Z,,MEMBER,,"- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.
","{'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50415/reactions', 'total_count': 0, '+1': 0, '-1': 0, 'laugh': 0, 'hooray': 0, 'confused': 0, 'heart': 0, 'rocket': 0, 'eyes': 0}",https://api.github.com/repos/pandas-dev/pandas/issues/50415/timeline,,,False,"{'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50415', 'html_url': 'https://github.com/pandas-dev/pandas/pull/50415', 'diff_url': 'https://github.com/pandas-dev/pandas/pull/50415.diff', 'patch_url': 'https://github.com/pandas-dev/pandas/pull/50415.patch', 'merged_at': None}"
7,https://api.github.com/repos/pandas-dev/pandas/issues/50414,https://api.github.com/repos/pandas-dev/pandas,https://api.github.com/repos/pandas-dev/pandas/issues/50414/labels{/name},https://api.github.com/repos/pandas-dev/pandas/issues/50414/comments,https://api.github.com/repos/pandas-dev/pandas/issues/50414/events,https://github.com/pandas-dev/pandas/pull/50414,1509536828,PR_kwDOAA0YD85GIVj6,50414,BUG: Grouper specified by key not regarded as in-axis,"{'login': 'rhshadrach', 'id': 45562402, 'node_id': 'MDQ6VXNlcjQ1NTYyNDAy', 'avatar_url': 'https://avatars.githubusercontent.com/u/45562402?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/rhshadrach', 'html_url': 'https://github.com/rhshadrach', 'followers_url': 'https://api.github.com/users/rhshadrach/followers', 'following_url': 'https://api.github.com/users/rhshadrach/following{/other_user}', 'gists_url': 'https://api.github.com/users/rhshadrach/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/rhshadrach/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/rhshadrach/subscriptions', 'organizations_url': 'https://api.github.com/users/rhshadrach/orgs', 'repos_url': 'https://api.github.com/users/rhshadrach/repos', 'events_url': 'https://api.github.com/users/rhshadrach/events{/privacy}', 'received_events_url': 'https://api.github.com/users/rhshadrach/received_events', 'type': 'User', 'site_admin': False}","[{'id': 76811, 'node_id': 'MDU6TGFiZWw3NjgxMQ==', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Bug', 'name': 'Bug', 'color': 'e10c02', 'default': False, 'description': None}, {'id': 233160, 'node_id': 'MDU6TGFiZWwyMzMxNjA=', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Groupby', 'name': 'Groupby', 'color': '729FCF', 'default': False, 'description': None}]",open,False,,[],"{'url': 'https://api.github.com/repos/pandas-dev/pandas/milestones/42', 'html_url': 'https://github.com/pandas-dev/pandas/milestone/42', 'labels_url': 'https://api.github.com/repos/pandas-dev/pandas/milestones/42/labels', 'id': 1945020, 'node_id': 'MDk6TWlsZXN0b25lMTk0NTAyMA==', 'number': 42, 'title': '2.0', 'description': '', 'creator': {'login': 'jreback', 'id': 953992, 'node_id': 'MDQ6VXNlcjk1Mzk5Mg==', 'avatar_url': 'https://avatars.githubusercontent.com/u/953992?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/jreback', 'html_url': 'https://github.com/jreback', 'followers_url': 'https://api.github.com/users/jreback/followers', 'following_url': 'https://api.github.com/users/jreback/following{/other_user}', 'gists_url': 'https://api.github.com/users/jreback/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/jreback/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/jreback/subscriptions', 'organizations_url': 'https://api.github.com/users/jreback/orgs', 'repos_url': 'https://api.github.com/users/jreback/repos', 'events_url': 'https://api.github.com/users/jreback/events{/privacy}', 'received_events_url': 'https://api.github.com/users/jreback/received_events', 'type': 'User', 'site_admin': False}, 'open_issues': 53, 'closed_issues': 851, 'state': 'open', 'created_at': '2016-08-16T13:21:24Z', 'updated_at': '2022-12-24T10:01:12Z', 'due_on': '2023-01-31T08:00:00Z', 'closed_at': None}",0,2022-12-23T16:34:14Z,2022-12-23T16:35:44Z,,MEMBER,,"- [x] closes #50413 (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Found this working on #46944. It was a case where we add to exclusions but consider the grouping as not in-axis. This creates a subtle difference in `_obj_with_exclusions` vs `_selected_obj`.","{'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50414/reactions', 'total_count': 0, '+1': 0, '-1': 0, 'laugh': 0, 'hooray': 0, 'confused': 0, 'heart': 0, 'rocket': 0, 'eyes': 0}",https://api.github.com/repos/pandas-dev/pandas/issues/50414/timeline,,,False,"{'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50414', 'html_url': 'https://github.com/pandas-dev/pandas/pull/50414', 'diff_url': 'https://github.com/pandas-dev/pandas/pull/50414.diff', 'patch_url': 'https://github.com/pandas-dev/pandas/pull/50414.patch', 'merged_at': None}"
8,https://api.github.com/repos/pandas-dev/pandas/issues/50413,https://api.github.com/repos/pandas-dev/pandas,https://api.github.com/repos/pandas-dev/pandas/issues/50413/labels{/name},https://api.github.com/repos/pandas-dev/pandas/issues/50413/comments,https://api.github.com/repos/pandas-dev/pandas/issues/50413/events,https://github.com/pandas-dev/pandas/issues/50413,1509519854,I_kwDOAA0YD85Z-XHu,50413,BUG: Grouper specified by key not regarded as in-axis,"{'login': 'rhshadrach', 'id': 45562402, 'node_id': 'MDQ6VXNlcjQ1NTYyNDAy', 'avatar_url': 'https://avatars.githubusercontent.com/u/45562402?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/rhshadrach', 'html_url': 'https://github.com/rhshadrach', 'followers_url': 'https://api.github.com/users/rhshadrach/followers', 'following_url': 'https://api.github.com/users/rhshadrach/following{/other_user}', 'gists_url': 'https://api.github.com/users/rhshadrach/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/rhshadrach/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/rhshadrach/subscriptions', 'organizations_url': 'https://api.github.com/users/rhshadrach/orgs', 'repos_url': 'https://api.github.com/users/rhshadrach/repos', 'events_url': 'https://api.github.com/users/rhshadrach/events{/privacy}', 'received_events_url': 'https://api.github.com/users/rhshadrach/received_events', 'type': 'User', 'site_admin': False}","[{'id': 76811, 'node_id': 'MDU6TGFiZWw3NjgxMQ==', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Bug', 'name': 'Bug', 'color': 'e10c02', 'default': False, 'description': None}, {'id': 233160, 'node_id': 'MDU6TGFiZWwyMzMxNjA=', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Groupby', 'name': 'Groupby', 'color': '729FCF', 'default': False, 'description': None}]",open,False,"{'login': 'rhshadrach', 'id': 45562402, 'node_id': 'MDQ6VXNlcjQ1NTYyNDAy', 'avatar_url': 'https://avatars.githubusercontent.com/u/45562402?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/rhshadrach', 'html_url': 'https://github.com/rhshadrach', 'followers_url': 'https://api.github.com/users/rhshadrach/followers', 'following_url': 'https://api.github.com/users/rhshadrach/following{/other_user}', 'gists_url': 'https://api.github.com/users/rhshadrach/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/rhshadrach/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/rhshadrach/subscriptions', 'organizations_url': 'https://api.github.com/users/rhshadrach/orgs', 'repos_url': 'https://api.github.com/users/rhshadrach/repos', 'events_url': 'https://api.github.com/users/rhshadrach/events{/privacy}', 'received_events_url': 'https://api.github.com/users/rhshadrach/received_events', 'type': 'User', 'site_admin': False}","[{'login': 'rhshadrach', 'id': 45562402, 'node_id': 'MDQ6VXNlcjQ1NTYyNDAy', 'avatar_url': 'https://avatars.githubusercontent.com/u/45562402?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/rhshadrach', 'html_url': 'https://github.com/rhshadrach', 'followers_url': 'https://api.github.com/users/rhshadrach/followers', 'following_url': 'https://api.github.com/users/rhshadrach/following{/other_user}', 'gists_url': 'https://api.github.com/users/rhshadrach/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/rhshadrach/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/rhshadrach/subscriptions', 'organizations_url': 'https://api.github.com/users/rhshadrach/orgs', 'repos_url': 'https://api.github.com/users/rhshadrach/repos', 'events_url': 'https://api.github.com/users/rhshadrach/events{/privacy}', 'received_events_url': 'https://api.github.com/users/rhshadrach/received_events', 'type': 'User', 'site_admin': False}]",,0,2022-12-23T16:17:28Z,2022-12-23T16:17:28Z,,MEMBER,,"Currently using `as_index=False` only includes in-axis groupers (see #49519). Defining a Grouper by the `key` argument is in-axis, but not regarded as such.

```
df = pd.DataFrame({'a': [1, 1, 2], 'b': [1, 1, 2], 'c': [3, 4, 5]})
gb = df.groupby([pd.Grouper(key='a'), pd.Grouper(key='b')], as_index=False)
result = gb.sum()
print(result)
#    c
# 0  7
# 1  5

gb2 = df.groupby(['a', 'b'], as_index=False)
result2 = gb2.sum()
print(result2)
#    a  b  c
# 0  1  1  7
# 1  2  2  5
```

The 1st case above should match the output of the 2nd case.","{'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50413/reactions', 'total_count': 0, '+1': 0, '-1': 0, 'laugh': 0, 'hooray': 0, 'confused': 0, 'heart': 0, 'rocket': 0, 'eyes': 0}",https://api.github.com/repos/pandas-dev/pandas/issues/50413/timeline,,,,
9,https://api.github.com/repos/pandas-dev/pandas/issues/50412,https://api.github.com/repos/pandas-dev/pandas,https://api.github.com/repos/pandas-dev/pandas/issues/50412/labels{/name},https://api.github.com/repos/pandas-dev/pandas/issues/50412/comments,https://api.github.com/repos/pandas-dev/pandas/issues/50412/events,https://github.com/pandas-dev/pandas/issues/50412,1509379729,I_kwDOAA0YD85Z906R,50412,BUG: inconsistent handling of exact=False case in to_datetime parsing,"{'login': 'jorisvandenbossche', 'id': 1020496, 'node_id': 'MDQ6VXNlcjEwMjA0OTY=', 'avatar_url': 'https://avatars.githubusercontent.com/u/1020496?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/jorisvandenbossche', 'html_url': 'https://github.com/jorisvandenbossche', 'followers_url': 'https://api.github.com/users/jorisvandenbossche/followers', 'following_url': 'https://api.github.com/users/jorisvandenbossche/following{/other_user}', 'gists_url': 'https://api.github.com/users/jorisvandenbossche/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/jorisvandenbossche/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/jorisvandenbossche/subscriptions', 'organizations_url': 'https://api.github.com/users/jorisvandenbossche/orgs', 'repos_url': 'https://api.github.com/users/jorisvandenbossche/repos', 'events_url': 'https://api.github.com/users/jorisvandenbossche/events{/privacy}', 'received_events_url': 'https://api.github.com/users/jorisvandenbossche/received_events', 'type': 'User', 'site_admin': False}","[{'id': 76811, 'node_id': 'MDU6TGFiZWw3NjgxMQ==', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Bug', 'name': 'Bug', 'color': 'e10c02', 'default': False, 'description': None}, {'id': 211840, 'node_id': 'MDU6TGFiZWwyMTE4NDA=', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Timeseries', 'name': 'Timeseries', 'color': 'AFEEEE', 'default': False, 'description': None}]",open,False,,[],,2,2022-12-23T14:00:25Z,2022-12-23T14:09:40Z,,MEMBER,,"From https://github.com/pandas-dev/pandas/issues/50411#issuecomment-1363972775

The documentation of `to_datetime` says:

> exact : bool, default True
>      Control how `format` is used:
>
>    - If :const:`True`, require an exact `format` match.
>    - If :const:`False`, allow the `format` to match anywhere in the target string.

Based on that, I would expect we don't use anything else in the string that doesn't match the format. But it seems this currently depends on the exact format being used:

```
>>> pd.to_datetime([""19/may/11 01:00:00""], format=""%d/%b/%y"", exact=False)
DatetimeIndex(['2011-05-19'], dtype='datetime64[ns]', freq=None)

>>> pd.to_datetime([""2011-05-19 01:00:00""], format=""%Y-%m-%d"", exact=False)
DatetimeIndex(['2011-05-19 01:00:00'], dtype='datetime64[ns]', freq=None)
```

(happens both on current main and on 1.5)","{'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50412/reactions', 'total_count': 0, '+1': 0, '-1': 0, 'laugh': 0, 'hooray': 0, 'confused': 0, 'heart': 0, 'rocket': 0, 'eyes': 0}",https://api.github.com/repos/pandas-dev/pandas/issues/50412/timeline,,,,
10,https://api.github.com/repos/pandas-dev/pandas/issues/50411,https://api.github.com/repos/pandas-dev/pandas,https://api.github.com/repos/pandas-dev/pandas/issues/50411/labels{/name},https://api.github.com/repos/pandas-dev/pandas/issues/50411/comments,https://api.github.com/repos/pandas-dev/pandas/issues/50411/events,https://github.com/pandas-dev/pandas/issues/50411,1509324259,I_kwDOAA0YD85Z9nXj,50411,Datetime parsing (PDEP-4): allow mixture of ISO formatted strings? ,"{'login': 'jorisvandenbossche', 'id': 1020496, 'node_id': 'MDQ6VXNlcjEwMjA0OTY=', 'avatar_url': 'https://avatars.githubusercontent.com/u/1020496?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/jorisvandenbossche', 'html_url': 'https://github.com/jorisvandenbossche', 'followers_url': 'https://api.github.com/users/jorisvandenbossche/followers', 'following_url': 'https://api.github.com/users/jorisvandenbossche/following{/other_user}', 'gists_url': 'https://api.github.com/users/jorisvandenbossche/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/jorisvandenbossche/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/jorisvandenbossche/subscriptions', 'organizations_url': 'https://api.github.com/users/jorisvandenbossche/orgs', 'repos_url': 'https://api.github.com/users/jorisvandenbossche/repos', 'events_url': 'https://api.github.com/users/jorisvandenbossche/events{/privacy}', 'received_events_url': 'https://api.github.com/users/jorisvandenbossche/received_events', 'type': 'User', 'site_admin': False}","[{'id': 211840, 'node_id': 'MDU6TGFiZWwyMTE4NDA=', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Timeseries', 'name': 'Timeseries', 'color': 'AFEEEE', 'default': False, 'description': None}]",open,False,,[],"{'url': 'https://api.github.com/repos/pandas-dev/pandas/milestones/42', 'html_url': 'https://github.com/pandas-dev/pandas/milestone/42', 'labels_url': 'https://api.github.com/repos/pandas-dev/pandas/milestones/42/labels', 'id': 1945020, 'node_id': 'MDk6TWlsZXN0b25lMTk0NTAyMA==', 'number': 42, 'title': '2.0', 'description': '', 'creator': {'login': 'jreback', 'id': 953992, 'node_id': 'MDQ6VXNlcjk1Mzk5Mg==', 'avatar_url': 'https://avatars.githubusercontent.com/u/953992?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/jreback', 'html_url': 'https://github.com/jreback', 'followers_url': 'https://api.github.com/users/jreback/followers', 'following_url': 'https://api.github.com/users/jreback/following{/other_user}', 'gists_url': 'https://api.github.com/users/jreback/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/jreback/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/jreback/subscriptions', 'organizations_url': 'https://api.github.com/users/jreback/orgs', 'repos_url': 'https://api.github.com/users/jreback/repos', 'events_url': 'https://api.github.com/users/jreback/events{/privacy}', 'received_events_url': 'https://api.github.com/users/jreback/received_events', 'type': 'User', 'site_admin': False}, 'open_issues': 53, 'closed_issues': 851, 'state': 'open', 'created_at': '2016-08-16T13:21:24Z', 'updated_at': '2022-12-24T10:01:12Z', 'due_on': '2023-01-31T08:00:00Z', 'closed_at': None}",9,2022-12-23T13:00:34Z,2022-12-23T18:48:32Z,,MEMBER,,"In pandas < 2, parsing datetime strings that are all strictly ISO formatted strings (so there is no ambiguity) but have slightly different formatting because of a different resolution, works fine:

```
>>> pd.to_datetime([""2022-01-01T09:00:00"", ""2022-01-02T09:00:00.123""])
DatetimeIndex(['2022-01-01 09:00:00', '2022-01-02 09:00:00.123000'], dtype='datetime64[ns]', freq=None)
```

With the changes related to datetime parsing (PDEP-4), this will now infer the format from the first, and then parsing the second fails:

```
>>> pd.to_datetime([""2022-01-01T09:00:00"", ""2022-01-02T09:00:00.123""])
...
ValueError: time data ""2022-01-02T09:00:00.123"" at position 1 doesn't match format ""%Y-%m-%dT%H:%M:%S""
```

This is of course expected and can be explained with the changes in behaviour. But I do wonder if we want to allow some way to still parse such data in a fast way. 

For this specific case, you can't specify a format since it is inconsistent. And before, this was actually fast because it took a fast ISO parsing path. 
With the current dev version of pandas, I don't see any way to parse this in a performant way? 

(this also came up in pyarrow's CI, it was an easy fix on our side to update the strings in the test, but still wanted to raise this as a specific case to consider, since with real world data you can't necessarily easily update your data)

cc @MarcoGorelli ","{'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50411/reactions', 'total_count': 0, '+1': 0, '-1': 0, 'laugh': 0, 'hooray': 0, 'confused': 0, 'heart': 0, 'rocket': 0, 'eyes': 0}",https://api.github.com/repos/pandas-dev/pandas/issues/50411/timeline,,,,
11,https://api.github.com/repos/pandas-dev/pandas/issues/50410,https://api.github.com/repos/pandas-dev/pandas,https://api.github.com/repos/pandas-dev/pandas/issues/50410/labels{/name},https://api.github.com/repos/pandas-dev/pandas/issues/50410/comments,https://api.github.com/repos/pandas-dev/pandas/issues/50410/events,https://github.com/pandas-dev/pandas/issues/50410,1509310192,I_kwDOAA0YD85Z9j7w,50410,Datetime parsing (PDEP-4): should time strings be inferred?,"{'login': 'jorisvandenbossche', 'id': 1020496, 'node_id': 'MDQ6VXNlcjEwMjA0OTY=', 'avatar_url': 'https://avatars.githubusercontent.com/u/1020496?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/jorisvandenbossche', 'html_url': 'https://github.com/jorisvandenbossche', 'followers_url': 'https://api.github.com/users/jorisvandenbossche/followers', 'following_url': 'https://api.github.com/users/jorisvandenbossche/following{/other_user}', 'gists_url': 'https://api.github.com/users/jorisvandenbossche/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/jorisvandenbossche/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/jorisvandenbossche/subscriptions', 'organizations_url': 'https://api.github.com/users/jorisvandenbossche/orgs', 'repos_url': 'https://api.github.com/users/jorisvandenbossche/repos', 'events_url': 'https://api.github.com/users/jorisvandenbossche/events{/privacy}', 'received_events_url': 'https://api.github.com/users/jorisvandenbossche/received_events', 'type': 'User', 'site_admin': False}","[{'id': 211840, 'node_id': 'MDU6TGFiZWwyMTE4NDA=', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Timeseries', 'name': 'Timeseries', 'color': 'AFEEEE', 'default': False, 'description': None}]",open,False,,[],"{'url': 'https://api.github.com/repos/pandas-dev/pandas/milestones/42', 'html_url': 'https://github.com/pandas-dev/pandas/milestone/42', 'labels_url': 'https://api.github.com/repos/pandas-dev/pandas/milestones/42/labels', 'id': 1945020, 'node_id': 'MDk6TWlsZXN0b25lMTk0NTAyMA==', 'number': 42, 'title': '2.0', 'description': '', 'creator': {'login': 'jreback', 'id': 953992, 'node_id': 'MDQ6VXNlcjk1Mzk5Mg==', 'avatar_url': 'https://avatars.githubusercontent.com/u/953992?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/jreback', 'html_url': 'https://github.com/jreback', 'followers_url': 'https://api.github.com/users/jreback/followers', 'following_url': 'https://api.github.com/users/jreback/following{/other_user}', 'gists_url': 'https://api.github.com/users/jreback/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/jreback/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/jreback/subscriptions', 'organizations_url': 'https://api.github.com/users/jreback/orgs', 'repos_url': 'https://api.github.com/users/jreback/repos', 'events_url': 'https://api.github.com/users/jreback/events{/privacy}', 'received_events_url': 'https://api.github.com/users/jreback/received_events', 'type': 'User', 'site_admin': False}, 'open_issues': 53, 'closed_issues': 851, 'state': 'open', 'created_at': '2016-08-16T13:21:24Z', 'updated_at': '2022-12-24T10:01:12Z', 'due_on': '2023-01-31T08:00:00Z', 'closed_at': None}",3,2022-12-23T12:43:42Z,2022-12-23T15:01:14Z,,MEMBER,,"Example with latest main (this now raises a warning, the actual result is the same as on released pandas):

```
>>> pd.to_datetime(['09:00', '09:30', '10:00'])
<stdin>:1: UserWarning: Could not infer format, so each element will be parsed individually by `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
DatetimeIndex(['2022-12-23 09:00:00', '2022-12-23 09:30:00',
               '2022-12-23 10:00:00'],
              dtype='datetime64[ns]', freq=None)
```

The above actually has a consistent format, but not one we can infer at the moment. Should we? 

(saw this warning in pyarrow's CI, we were using `to_datetime` to parse time strings into datetime.time objects, by directly accessing `.time()`, so the actual date didn't matter in that use case)

cc @MarcoGorelli 
","{'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50410/reactions', 'total_count': 0, '+1': 0, '-1': 0, 'laugh': 0, 'hooray': 0, 'confused': 0, 'heart': 0, 'rocket': 0, 'eyes': 0}",https://api.github.com/repos/pandas-dev/pandas/issues/50410/timeline,,,,
12,https://api.github.com/repos/pandas-dev/pandas/issues/50409,https://api.github.com/repos/pandas-dev/pandas,https://api.github.com/repos/pandas-dev/pandas/issues/50409/labels{/name},https://api.github.com/repos/pandas-dev/pandas/issues/50409/comments,https://api.github.com/repos/pandas-dev/pandas/issues/50409/events,https://github.com/pandas-dev/pandas/issues/50409,1509146751,I_kwDOAA0YD85Z88B_,50409,BUG: Read_csv won't warn/skip bad lines when nrows is being used,"{'login': 'd2thebee', 'id': 79337886, 'node_id': 'MDQ6VXNlcjc5MzM3ODg2', 'avatar_url': 'https://avatars.githubusercontent.com/u/79337886?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/d2thebee', 'html_url': 'https://github.com/d2thebee', 'followers_url': 'https://api.github.com/users/d2thebee/followers', 'following_url': 'https://api.github.com/users/d2thebee/following{/other_user}', 'gists_url': 'https://api.github.com/users/d2thebee/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/d2thebee/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/d2thebee/subscriptions', 'organizations_url': 'https://api.github.com/users/d2thebee/orgs', 'repos_url': 'https://api.github.com/users/d2thebee/repos', 'events_url': 'https://api.github.com/users/d2thebee/events{/privacy}', 'received_events_url': 'https://api.github.com/users/d2thebee/received_events', 'type': 'User', 'site_admin': False}","[{'id': 76811, 'node_id': 'MDU6TGFiZWw3NjgxMQ==', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Bug', 'name': 'Bug', 'color': 'e10c02', 'default': False, 'description': None}, {'id': 47229171, 'node_id': 'MDU6TGFiZWw0NzIyOTE3MQ==', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/IO%20CSV', 'name': 'IO CSV', 'color': '5319e7', 'default': False, 'description': 'read_csv, to_csv'}, {'id': 307649777, 'node_id': 'MDU6TGFiZWwzMDc2NDk3Nzc=', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Needs%20Info', 'name': 'Needs Info', 'color': '207de5', 'default': False, 'description': 'Clarification about behavior needed to assess issue'}]",open,False,,[],,1,2022-12-23T10:04:33Z,2022-12-23T15:44:59Z,,NONE,,"### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the main branch of pandas.


### Reproducible Example

```python
import pandas as pd
data = pd.read_csv('file.csv', sep='¬', encoding = 'ISO-8859-1', engine='python', on_bad_lines='warn', nrows=2845060)
```


### Issue Description

Read_csv with on_bad_lines AND nrows at the same time appears to not follow the correct on_bad_lines logic. When I run the above example without the nrows argument, I get the following warning:

Skipping line 2845058: '¬' expected after '""'

But running with nrows argument included, it errors out with the following message: (same happens when I use on_bad_lines='skip' instead of 'warn')

Error: '¬' expected after '""'

### Expected Behavior

The on_bad_lines logic should continue to work when using nrows so I should see the 'Skipping line' warning.

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 8dab54d6573f7186ff0c3b6364d5e4dd635ff3e7
python           : 3.10.6.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.19044
machine          : AMD64
processor        : Intel64 Family 6 Model 154 Stepping 3, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : English_United Kingdom.1252

pandas           : 1.5.2
numpy            : 1.23.2
pytz             : 2022.1
dateutil         : 2.8.2
setuptools       : 63.2.0
pip              : 22.2.2
Cython           : None
pytest           : None
hypothesis       : None
...
xlrd             : 2.0.1
xlwt             : None
zstandard        : None
tzdata           : None

</details>
","{'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50409/reactions', 'total_count': 0, '+1': 0, '-1': 0, 'laugh': 0, 'hooray': 0, 'confused': 0, 'heart': 0, 'rocket': 0, 'eyes': 0}",https://api.github.com/repos/pandas-dev/pandas/issues/50409/timeline,,,,
13,https://api.github.com/repos/pandas-dev/pandas/issues/50408,https://api.github.com/repos/pandas-dev/pandas,https://api.github.com/repos/pandas-dev/pandas/issues/50408/labels{/name},https://api.github.com/repos/pandas-dev/pandas/issues/50408/comments,https://api.github.com/repos/pandas-dev/pandas/issues/50408/events,https://github.com/pandas-dev/pandas/pull/50408,1509071505,PR_kwDOAA0YD85GGv7x,50408,STYLE: prohibit parsing exeption messages,"{'login': 'luke396', 'id': 77875500, 'node_id': 'MDQ6VXNlcjc3ODc1NTAw', 'avatar_url': 'https://avatars.githubusercontent.com/u/77875500?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/luke396', 'html_url': 'https://github.com/luke396', 'followers_url': 'https://api.github.com/users/luke396/followers', 'following_url': 'https://api.github.com/users/luke396/following{/other_user}', 'gists_url': 'https://api.github.com/users/luke396/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/luke396/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/luke396/subscriptions', 'organizations_url': 'https://api.github.com/users/luke396/orgs', 'repos_url': 'https://api.github.com/users/luke396/repos', 'events_url': 'https://api.github.com/users/luke396/events{/privacy}', 'received_events_url': 'https://api.github.com/users/luke396/received_events', 'type': 'User', 'site_admin': False}",[],open,False,,[],,0,2022-12-23T08:50:43Z,2022-12-24T11:56:27Z,,CONTRIBUTOR,,"to fix #50353. 

I'm not quite sure I'm doing it right, so I've just tackled some, and  deal them little by little. Welcome some code review and comments to help me determine that I'm in the right way!","{'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50408/reactions', 'total_count': 0, '+1': 0, '-1': 0, 'laugh': 0, 'hooray': 0, 'confused': 0, 'heart': 0, 'rocket': 0, 'eyes': 0}",https://api.github.com/repos/pandas-dev/pandas/issues/50408/timeline,,,False,"{'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50408', 'html_url': 'https://github.com/pandas-dev/pandas/pull/50408', 'diff_url': 'https://github.com/pandas-dev/pandas/pull/50408.diff', 'patch_url': 'https://github.com/pandas-dev/pandas/pull/50408.patch', 'merged_at': None}"
14,https://api.github.com/repos/pandas-dev/pandas/issues/50407,https://api.github.com/repos/pandas-dev/pandas,https://api.github.com/repos/pandas-dev/pandas/issues/50407/labels{/name},https://api.github.com/repos/pandas-dev/pandas/issues/50407/comments,https://api.github.com/repos/pandas-dev/pandas/issues/50407/events,https://github.com/pandas-dev/pandas/issues/50407,1508891678,I_kwDOAA0YD85Z79we,50407,DEPR: Unnecessary kwargs in groupby ops,"{'login': 'rhshadrach', 'id': 45562402, 'node_id': 'MDQ6VXNlcjQ1NTYyNDAy', 'avatar_url': 'https://avatars.githubusercontent.com/u/45562402?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/rhshadrach', 'html_url': 'https://github.com/rhshadrach', 'followers_url': 'https://api.github.com/users/rhshadrach/followers', 'following_url': 'https://api.github.com/users/rhshadrach/following{/other_user}', 'gists_url': 'https://api.github.com/users/rhshadrach/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/rhshadrach/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/rhshadrach/subscriptions', 'organizations_url': 'https://api.github.com/users/rhshadrach/orgs', 'repos_url': 'https://api.github.com/users/rhshadrach/repos', 'events_url': 'https://api.github.com/users/rhshadrach/events{/privacy}', 'received_events_url': 'https://api.github.com/users/rhshadrach/received_events', 'type': 'User', 'site_admin': False}","[{'id': 233160, 'node_id': 'MDU6TGFiZWwyMzMxNjA=', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Groupby', 'name': 'Groupby', 'color': '729FCF', 'default': False, 'description': None}, {'id': 87485152, 'node_id': 'MDU6TGFiZWw4NzQ4NTE1Mg==', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Deprecate', 'name': 'Deprecate', 'color': '5319e7', 'default': False, 'description': 'Functionality to remove in pandas'}]",open,False,,[],,0,2022-12-23T05:03:25Z,2022-12-23T05:03:25Z,,MEMBER,,"Some groupby ops have args and/or kwargs unnecessarily.

 - cummax
 - cummin
 - cumsum
 - cumprod
 - take
 - skew

These should be deprecated. For the cum* methods, the kwargs that are used should be added to the function signature (e.g. `skipna`).","{'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50407/reactions', 'total_count': 0, '+1': 0, '-1': 0, 'laugh': 0, 'hooray': 0, 'confused': 0, 'heart': 0, 'rocket': 0, 'eyes': 0}",https://api.github.com/repos/pandas-dev/pandas/issues/50407/timeline,,,,
15,https://api.github.com/repos/pandas-dev/pandas/issues/50406,https://api.github.com/repos/pandas-dev/pandas,https://api.github.com/repos/pandas-dev/pandas/issues/50406/labels{/name},https://api.github.com/repos/pandas-dev/pandas/issues/50406/comments,https://api.github.com/repos/pandas-dev/pandas/issues/50406/events,https://github.com/pandas-dev/pandas/issues/50406,1508887523,I_kwDOAA0YD85Z78vj,50406,ENH: Use explicit arguments in DataFrameGroupBy.describe,"{'login': 'rhshadrach', 'id': 45562402, 'node_id': 'MDQ6VXNlcjQ1NTYyNDAy', 'avatar_url': 'https://avatars.githubusercontent.com/u/45562402?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/rhshadrach', 'html_url': 'https://github.com/rhshadrach', 'followers_url': 'https://api.github.com/users/rhshadrach/followers', 'following_url': 'https://api.github.com/users/rhshadrach/following{/other_user}', 'gists_url': 'https://api.github.com/users/rhshadrach/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/rhshadrach/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/rhshadrach/subscriptions', 'organizations_url': 'https://api.github.com/users/rhshadrach/orgs', 'repos_url': 'https://api.github.com/users/rhshadrach/repos', 'events_url': 'https://api.github.com/users/rhshadrach/events{/privacy}', 'received_events_url': 'https://api.github.com/users/rhshadrach/received_events', 'type': 'User', 'site_admin': False}","[{'id': 76812, 'node_id': 'MDU6TGFiZWw3NjgxMg==', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Enhancement', 'name': 'Enhancement', 'color': '4E9A06', 'default': False, 'description': None}, {'id': 233160, 'node_id': 'MDU6TGFiZWwyMzMxNjA=', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Groupby', 'name': 'Groupby', 'color': '729FCF', 'default': False, 'description': None}, {'id': 717120670, 'node_id': 'MDU6TGFiZWw3MTcxMjA2NzA=', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/good%20first%20issue', 'name': 'good first issue', 'color': '0e8a16', 'default': True, 'description': None}]",open,False,"{'login': 'dullibri', 'id': 9106258, 'node_id': 'MDQ6VXNlcjkxMDYyNTg=', 'avatar_url': 'https://avatars.githubusercontent.com/u/9106258?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/dullibri', 'html_url': 'https://github.com/dullibri', 'followers_url': 'https://api.github.com/users/dullibri/followers', 'following_url': 'https://api.github.com/users/dullibri/following{/other_user}', 'gists_url': 'https://api.github.com/users/dullibri/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/dullibri/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/dullibri/subscriptions', 'organizations_url': 'https://api.github.com/users/dullibri/orgs', 'repos_url': 'https://api.github.com/users/dullibri/repos', 'events_url': 'https://api.github.com/users/dullibri/events{/privacy}', 'received_events_url': 'https://api.github.com/users/dullibri/received_events', 'type': 'User', 'site_admin': False}","[{'login': 'dullibri', 'id': 9106258, 'node_id': 'MDQ6VXNlcjkxMDYyNTg=', 'avatar_url': 'https://avatars.githubusercontent.com/u/9106258?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/dullibri', 'html_url': 'https://github.com/dullibri', 'followers_url': 'https://api.github.com/users/dullibri/followers', 'following_url': 'https://api.github.com/users/dullibri/following{/other_user}', 'gists_url': 'https://api.github.com/users/dullibri/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/dullibri/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/dullibri/subscriptions', 'organizations_url': 'https://api.github.com/users/dullibri/orgs', 'repos_url': 'https://api.github.com/users/dullibri/repos', 'events_url': 'https://api.github.com/users/dullibri/events{/privacy}', 'received_events_url': 'https://api.github.com/users/dullibri/received_events', 'type': 'User', 'site_admin': False}]",,1,2022-12-23T04:57:41Z,2022-12-23T09:20:56Z,,MEMBER,,"Currently DataFrameGroupBy.describe uses `**kwargs` instead of explicit arguments. These eventually get passed to `DataFrame.describe`. We should use explicit arguments, those from `DataFrame.describe`, instead.","{'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50406/reactions', 'total_count': 0, '+1': 0, '-1': 0, 'laugh': 0, 'hooray': 0, 'confused': 0, 'heart': 0, 'rocket': 0, 'eyes': 0}",https://api.github.com/repos/pandas-dev/pandas/issues/50406/timeline,,,,
16,https://api.github.com/repos/pandas-dev/pandas/issues/50405,https://api.github.com/repos/pandas-dev/pandas,https://api.github.com/repos/pandas-dev/pandas/issues/50405/labels{/name},https://api.github.com/repos/pandas-dev/pandas/issues/50405/comments,https://api.github.com/repos/pandas-dev/pandas/issues/50405/events,https://github.com/pandas-dev/pandas/issues/50405,1508864485,I_kwDOAA0YD85Z73Hl,50405,DEPR: axis argument in groupby ops,"{'login': 'rhshadrach', 'id': 45562402, 'node_id': 'MDQ6VXNlcjQ1NTYyNDAy', 'avatar_url': 'https://avatars.githubusercontent.com/u/45562402?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/rhshadrach', 'html_url': 'https://github.com/rhshadrach', 'followers_url': 'https://api.github.com/users/rhshadrach/followers', 'following_url': 'https://api.github.com/users/rhshadrach/following{/other_user}', 'gists_url': 'https://api.github.com/users/rhshadrach/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/rhshadrach/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/rhshadrach/subscriptions', 'organizations_url': 'https://api.github.com/users/rhshadrach/orgs', 'repos_url': 'https://api.github.com/users/rhshadrach/repos', 'events_url': 'https://api.github.com/users/rhshadrach/events{/privacy}', 'received_events_url': 'https://api.github.com/users/rhshadrach/received_events', 'type': 'User', 'site_admin': False}","[{'id': 233160, 'node_id': 'MDU6TGFiZWwyMzMxNjA=', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Groupby', 'name': 'Groupby', 'color': '729FCF', 'default': False, 'description': None}, {'id': 87485152, 'node_id': 'MDU6TGFiZWw4NzQ4NTE1Mg==', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Deprecate', 'name': 'Deprecate', 'color': '5319e7', 'default': False, 'description': 'Functionality to remove in pandas'}, {'id': 219960758, 'node_id': 'MDU6TGFiZWwyMTk5NjA3NTg=', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Needs%20Discussion', 'name': 'Needs Discussion', 'color': '207de5', 'default': False, 'description': 'Requires discussion from core team before further action'}]",open,False,,[],,0,2022-12-23T04:19:08Z,2022-12-23T04:19:19Z,,MEMBER,,"Some groupby methods have an `axis` argument in addition to the `axis` argument of `DataFrame.groupby`. When these two axis arguments do not agree (e.g. `df.groupby(..., axis=0).skew(axis=1)`), it provides the same operation that can be achieved more efficiently without using groupby. More explicitly, 

 - `df.groupby(..., axis=1).op(axis=0)` will  operate column by column, the same as `DataFrame.op(axis=0)`
 - `df.groupby(..., axis=0).op(axis=1)` will operate row by row, the same as `DataFrame.op(axis=1)`

There is some reshaping that is necessary to get exact equality (see the code below) because groupby shapes the indices in a specific way.

In addition to this, various groupby ops either fail or effectively ignore argument (it is used in the code, but in such a way that it does not influence the result).

I propose we deprecate the `axis` argument across groupby ops. For those ops where it is necessary, the future behavior will have the op's `axis` value the same as that provided to groupby. In other words, all ops either act as `df.groupby(..., axis=0).op(axis=0)` or `df.groupby(..., axis=1).op(axis=1)`.

Code:

<details>
<summary>`df.groupby(..., axis=0).op(axis=1)`</summary>

```
import inspect

import pandas._testing as tm
from pandas.core.groupby.base import reduction_kernels, groupby_other_methods, transformation_kernels
from pandas.tests.groupby import get_groupby_method_args

size = 10
df = pd.DataFrame(
    {
        'a': np.random.randint(0, 3, size),
        'b': np.random.randint(0, 10, size),
        'c': np.random.randint(0, 10, size),
        'd': np.random.randint(0, 10, size),
    }
)
gb = df.groupby('a', axis=0)

kernels = reduction_kernels | groupby_other_methods | transformation_kernels
for kernel in kernels:
    if kernel in ('plot',):
        continue
    try:
        op = getattr(gb, kernel)
        if not callable(op):
            continue
        arguments = inspect.signature(op).parameters.keys()
        if 'axis' not in arguments:
            continue
        args = get_groupby_method_args(kernel, df)
        result = op(*args, axis=1)
        
        if kernel in ('skew', 'idxmax', 'idxmin', 'fillna', 'take'):
            expected = getattr(df[['b', 'c', 'd']], kernel)(*args, axis=1)
        else:
            expected = getattr(df, kernel)(*args, axis=1)
    except Exception as err:
        print('Failed:', kernel, err)
    else:
        if kernel in ('skew', 'idxmax', 'idxmin'):
            expected = expected.to_frame('values')
        if kernel in ('diff', 'skew', 'idxmax', 'idxmin', 'take'):
            expected = expected.set_index(df['a'], append=True)
            expected = expected.swaplevel(0, 1).sort_index()
        if kernel in ('skew', 'idxmax', 'idxmin'):
            expected = expected['values'].rename(None)
        try:
            tm.assert_equal(result, expected)
        except Exception as err:
            print(kernel)
```

</details>

<details>
<summary>`df.groupby(..., axis=1).op(axis=0)`</summary>

```
import inspect

import pandas._testing as tm
from pandas.core.groupby.base import reduction_kernels, groupby_other_methods, transformation_kernels
from pandas.tests.groupby import get_groupby_method_args

size = 10
df = pd.DataFrame(
    {
        'a': np.random.randint(0, 3, size),
        'b': np.random.randint(0, 10, size),
        'c': np.random.randint(0, 10, size),
        'd': np.random.randint(0, 10, size),
    }
)
gb = df.groupby([0, 0, 0, 1], axis=1)

kernels = reduction_kernels | groupby_other_methods | transformation_kernels
for kernel in kernels:
    if kernel in ('plot',):
        continue
    if kernel in ('rank', 'cummax', 'cummin', 'cumprod', 'cumsum', 'shift', 'diff', 'pct_change'):
        # Already ignores the axis argument
        continue
    if kernel in ('idxmin', 'idxmax', 'skew'):
        # Raises when using .groupby(..., axis=1).op(axis=0)
        continue
    try:
        op = getattr(gb, kernel)
        if not callable(op):
            continue
        arguments = inspect.signature(op).parameters.keys()
        if 'axis' not in arguments:
            continue
        args = get_groupby_method_args(kernel, df)
        result = op(*args, axis=0)
        expected = getattr(df, kernel)(*args, axis=0)
    except Exception as err:
        print('Failed:', kernel, err)
    else:
        try:
            tm.assert_equal(result, expected)
        except Exception as err:
            print(kernel)
```

</details>","{'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50405/reactions', 'total_count': 0, '+1': 0, '-1': 0, 'laugh': 0, 'hooray': 0, 'confused': 0, 'heart': 0, 'rocket': 0, 'eyes': 0}",https://api.github.com/repos/pandas-dev/pandas/issues/50405/timeline,,,,
17,https://api.github.com/repos/pandas-dev/pandas/issues/50402,https://api.github.com/repos/pandas-dev/pandas,https://api.github.com/repos/pandas-dev/pandas/issues/50402/labels{/name},https://api.github.com/repos/pandas-dev/pandas/issues/50402/comments,https://api.github.com/repos/pandas-dev/pandas/issues/50402/events,https://github.com/pandas-dev/pandas/issues/50402,1508558921,I_kwDOAA0YD85Z6shJ,50402,BUG: dictionary groupby aggregation fails in case there're duplicated values in MultiIndex columns,"{'login': 'dchigarev', 'id': 62142979, 'node_id': 'MDQ6VXNlcjYyMTQyOTc5', 'avatar_url': 'https://avatars.githubusercontent.com/u/62142979?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/dchigarev', 'html_url': 'https://github.com/dchigarev', 'followers_url': 'https://api.github.com/users/dchigarev/followers', 'following_url': 'https://api.github.com/users/dchigarev/following{/other_user}', 'gists_url': 'https://api.github.com/users/dchigarev/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/dchigarev/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/dchigarev/subscriptions', 'organizations_url': 'https://api.github.com/users/dchigarev/orgs', 'repos_url': 'https://api.github.com/users/dchigarev/repos', 'events_url': 'https://api.github.com/users/dchigarev/events{/privacy}', 'received_events_url': 'https://api.github.com/users/dchigarev/received_events', 'type': 'User', 'site_admin': False}","[{'id': 76811, 'node_id': 'MDU6TGFiZWw3NjgxMQ==', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Bug', 'name': 'Bug', 'color': 'e10c02', 'default': False, 'description': None}, {'id': 233160, 'node_id': 'MDU6TGFiZWwyMzMxNjA=', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Groupby', 'name': 'Groupby', 'color': '729FCF', 'default': False, 'description': None}]",open,False,,[],,1,2022-12-22T21:20:00Z,2022-12-22T21:54:38Z,,NONE,,"### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the main branch of pandas.


### Reproducible Example

```python
import pandas

df = pandas.DataFrame(
    [[1, 2, 3], [4, 5, 6]],
    columns=pandas.MultiIndex.from_tuples([(""a"", 0), (""b"", 0), (""b"", 1)]),
)

res = df.groupby(by=[1, 2]).agg({(""a"", 0): ""sum""})  # works fine
print(res)

df.columns = pandas.MultiIndex.from_tuples([(""a"", 0), (""b"", 0), (""b"", 0)])
res = df.groupby(by=[1, 2]).agg(
    {(""a"", 0): ""sum""}
)  # AttributeError: 'Series' object has no attribute 'columns'
print(res)
```


### Issue Description

<details><summary>Reproducer's traceback</summary>

```python-traceback
Traceback (most recent call last):
  File ""t2.py"", line 12, in <module>
    res = df.groupby(by=[1, 2]).agg({(""a"", 0): ""sum""}) # AttributeError: 'Series' object has no attribute 'columns'
  File ""/localdisk/dchigare/miniconda3/envs/modin/lib/python3.8/site-packages/pandas/core/groupby/generic.py"", line 894, in aggregate
    result = op.agg()
  File ""/localdisk/dchigare/miniconda3/envs/modin/lib/python3.8/site-packages/pandas/core/apply.py"", line 169, in agg
    return self.agg_dict_like()
  File ""/localdisk/dchigare/miniconda3/envs/modin/lib/python3.8/site-packages/pandas/core/apply.py"", line 486, in agg_dict_like
    results = {
  File ""/localdisk/dchigare/miniconda3/envs/modin/lib/python3.8/site-packages/pandas/core/apply.py"", line 487, in <dictcomp>
    key: obj._gotitem(key, ndim=1).agg(how) for key, how in arg.items()
  File ""/localdisk/dchigare/miniconda3/envs/modin/lib/python3.8/site-packages/pandas/core/groupby/generic.py"", line 275, in aggregate
    return getattr(self, func)(*args, **kwargs)
  File ""/localdisk/dchigare/miniconda3/envs/modin/lib/python3.8/site-packages/pandas/core/groupby/groupby.py"", line 2425, in sum
    result = self._agg_general(
  File ""/localdisk/dchigare/miniconda3/envs/modin/lib/python3.8/site-packages/pandas/core/groupby/groupby.py"", line 1683, in _agg_general
    result = self._cython_agg_general(
  File ""/localdisk/dchigare/miniconda3/envs/modin/lib/python3.8/site-packages/pandas/core/groupby/groupby.py"", line 1743, in _cython_agg_general
    numeric_only_bool = self._resolve_numeric_only(how, numeric_only, axis=0)
  File ""/localdisk/dchigare/miniconda3/envs/modin/lib/python3.8/site-packages/pandas/core/groupby/groupby.py"", line 1347, in _resolve_numeric_only
    if len(obj.columns) and not len(check.columns) and not obj.empty:
  File ""/localdisk/dchigare/miniconda3/envs/modin/lib/python3.8/site-packages/pandas/core/generic.py"", line 5902, in __getattr__
    return object.__getattribute__(self, name)
AttributeError: 'Series' object has no attribute 'columns'
```

</details>

### Expected Behavior

Groupby on a frame with duplicated non-MultiIndex values works fine, I would expect the MultiIndex case to work the same.
```python
import pandas

df = pandas.DataFrame(
    [[1, 2, 3], [4, 5, 6]],
    columns=pandas.Index([""a"", ""b"", ""b""]),
)

res = df.groupby(by=[1, 2]).agg({""a"": ""sum""})  # works fine
print(res)
```

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 8dab54d6573f7186ff0c3b6364d5e4dd635ff3e7
python           : 3.8.13.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.4.0-124-generic
Version          : #140-Ubuntu SMP Thu Aug 4 02:23:37 UTC 2022
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.5.2
numpy            : 1.23.3
pytz             : 2022.4
dateutil         : 2.8.2
setuptools       : 65.4.1
pip              : 22.2.2
Cython           : None
pytest           : 7.1.3
hypothesis       : None
sphinx           : 5.3.0
blosc            : None
feather          : 0.4.1
xlsxwriter       : None
lxml.etree       : 4.9.1
html5lib         : None
pymysql          : None
psycopg2         : 2.9.3
jinja2           : 3.1.2
IPython          : 8.4.0
pandas_datareader: None
bs4              : None
bottleneck       : None
brotli           :
fastparquet      : 0.8.3
fsspec           : 2022.8.2
gcsfs            : None
matplotlib       : 3.3.0
numba            : None
numexpr          : 2.8.3
odfpy            : None
openpyxl         : 3.0.10
pandas_gbq       : 0.17.9
pyarrow          : 9.0.0
pyreadstat       : None
pyxlsb           : None
s3fs             : 2022.8.2
scipy            : 1.9.1
snappy           : None
sqlalchemy       : 1.4.41
tables           : 3.7.0
tabulate         : None
xarray           : 2022.9.0
xlrd             : 2.0.1
xlwt             : None
zstandard        : None
tzdata           : None

</details>
","{'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50402/reactions', 'total_count': 0, '+1': 0, '-1': 0, 'laugh': 0, 'hooray': 0, 'confused': 0, 'heart': 0, 'rocket': 0, 'eyes': 0}",https://api.github.com/repos/pandas-dev/pandas/issues/50402/timeline,,,,
18,https://api.github.com/repos/pandas-dev/pandas/issues/50401,https://api.github.com/repos/pandas-dev/pandas,https://api.github.com/repos/pandas-dev/pandas/issues/50401/labels{/name},https://api.github.com/repos/pandas-dev/pandas/issues/50401/comments,https://api.github.com/repos/pandas-dev/pandas/issues/50401/events,https://github.com/pandas-dev/pandas/pull/50401,1508378234,PR_kwDOAA0YD85GEZUQ,50401,REF: restore _concat_managers_axis0,"{'login': 'jbrockmendel', 'id': 8078968, 'node_id': 'MDQ6VXNlcjgwNzg5Njg=', 'avatar_url': 'https://avatars.githubusercontent.com/u/8078968?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/jbrockmendel', 'html_url': 'https://github.com/jbrockmendel', 'followers_url': 'https://api.github.com/users/jbrockmendel/followers', 'following_url': 'https://api.github.com/users/jbrockmendel/following{/other_user}', 'gists_url': 'https://api.github.com/users/jbrockmendel/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/jbrockmendel/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/jbrockmendel/subscriptions', 'organizations_url': 'https://api.github.com/users/jbrockmendel/orgs', 'repos_url': 'https://api.github.com/users/jbrockmendel/repos', 'events_url': 'https://api.github.com/users/jbrockmendel/events{/privacy}', 'received_events_url': 'https://api.github.com/users/jbrockmendel/received_events', 'type': 'User', 'site_admin': False}",[],open,False,,[],,0,2022-12-22T18:35:11Z,2022-12-22T18:35:11Z,,MEMBER,,"It was removed in #47372 but removing it was not the point of that PR, just a side-effect.","{'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50401/reactions', 'total_count': 0, '+1': 0, '-1': 0, 'laugh': 0, 'hooray': 0, 'confused': 0, 'heart': 0, 'rocket': 0, 'eyes': 0}",https://api.github.com/repos/pandas-dev/pandas/issues/50401/timeline,,,False,"{'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50401', 'html_url': 'https://github.com/pandas-dev/pandas/pull/50401', 'diff_url': 'https://github.com/pandas-dev/pandas/pull/50401.diff', 'patch_url': 'https://github.com/pandas-dev/pandas/pull/50401.patch', 'merged_at': None}"
19,https://api.github.com/repos/pandas-dev/pandas/issues/50400,https://api.github.com/repos/pandas-dev/pandas,https://api.github.com/repos/pandas-dev/pandas/issues/50400/labels{/name},https://api.github.com/repos/pandas-dev/pandas/issues/50400/comments,https://api.github.com/repos/pandas-dev/pandas/issues/50400/events,https://github.com/pandas-dev/pandas/pull/50400,1508254094,PR_kwDOAA0YD85GD9HI,50400,BUG: Index with null value not serialized correctly to json,"{'login': 'lithomas1', 'id': 47963215, 'node_id': 'MDQ6VXNlcjQ3OTYzMjE1', 'avatar_url': 'https://avatars.githubusercontent.com/u/47963215?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/lithomas1', 'html_url': 'https://github.com/lithomas1', 'followers_url': 'https://api.github.com/users/lithomas1/followers', 'following_url': 'https://api.github.com/users/lithomas1/following{/other_user}', 'gists_url': 'https://api.github.com/users/lithomas1/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/lithomas1/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/lithomas1/subscriptions', 'organizations_url': 'https://api.github.com/users/lithomas1/orgs', 'repos_url': 'https://api.github.com/users/lithomas1/repos', 'events_url': 'https://api.github.com/users/lithomas1/events{/privacy}', 'received_events_url': 'https://api.github.com/users/lithomas1/received_events', 'type': 'User', 'site_admin': False}","[{'id': 76811, 'node_id': 'MDU6TGFiZWw3NjgxMQ==', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Bug', 'name': 'Bug', 'color': 'e10c02', 'default': False, 'description': None}, {'id': 49379259, 'node_id': 'MDU6TGFiZWw0OTM3OTI1OQ==', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/IO%20JSON', 'name': 'IO JSON', 'color': '207de5', 'default': False, 'description': 'read_json, to_json, json_normalize'}, {'id': 1218227310, 'node_id': 'MDU6TGFiZWwxMjE4MjI3MzEw', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Index', 'name': 'Index', 'color': 'e99695', 'default': False, 'description': 'Related to the Index class or subclasses'}]",open,False,,[],,0,2022-12-22T17:12:54Z,2022-12-23T18:43:20Z,,MEMBER,,"- [ ] closes #31801 (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Restores 0.25.3 behavior, which honestly feels a little weird to me, since it doesn't roundtrip. I'm not too familiar with JSON format, though (I guess the problem is that null can't be a key?).

Also serialize decimal nans correctly.
","{'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50400/reactions', 'total_count': 0, '+1': 0, '-1': 0, 'laugh': 0, 'hooray': 0, 'confused': 0, 'heart': 0, 'rocket': 0, 'eyes': 0}",https://api.github.com/repos/pandas-dev/pandas/issues/50400/timeline,,,False,"{'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50400', 'html_url': 'https://github.com/pandas-dev/pandas/pull/50400', 'diff_url': 'https://github.com/pandas-dev/pandas/pull/50400.diff', 'patch_url': 'https://github.com/pandas-dev/pandas/pull/50400.patch', 'merged_at': None}"
20,https://api.github.com/repos/pandas-dev/pandas/issues/50398,https://api.github.com/repos/pandas-dev/pandas,https://api.github.com/repos/pandas-dev/pandas/issues/50398/labels{/name},https://api.github.com/repos/pandas-dev/pandas/issues/50398/comments,https://api.github.com/repos/pandas-dev/pandas/issues/50398/events,https://github.com/pandas-dev/pandas/pull/50398,1508195560,PR_kwDOAA0YD85GDvwX,50398,"ENH: add math mode to formatter escape=""latex-math""","{'login': 'natmokval', 'id': 91160475, 'node_id': 'MDQ6VXNlcjkxMTYwNDc1', 'avatar_url': 'https://avatars.githubusercontent.com/u/91160475?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/natmokval', 'html_url': 'https://github.com/natmokval', 'followers_url': 'https://api.github.com/users/natmokval/followers', 'following_url': 'https://api.github.com/users/natmokval/following{/other_user}', 'gists_url': 'https://api.github.com/users/natmokval/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/natmokval/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/natmokval/subscriptions', 'organizations_url': 'https://api.github.com/users/natmokval/orgs', 'repos_url': 'https://api.github.com/users/natmokval/repos', 'events_url': 'https://api.github.com/users/natmokval/events{/privacy}', 'received_events_url': 'https://api.github.com/users/natmokval/received_events', 'type': 'User', 'site_admin': False}",[],open,False,,[],,0,2022-12-22T16:43:20Z,2022-12-23T10:42:56Z,,CONTRIBUTOR,,"- [x] closes #50040
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
 
Added `latex-math` option to `_str_escape` to avoid escaping `$`, `\[` and `\]`.","{'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50398/reactions', 'total_count': 0, '+1': 0, '-1': 0, 'laugh': 0, 'hooray': 0, 'confused': 0, 'heart': 0, 'rocket': 0, 'eyes': 0}",https://api.github.com/repos/pandas-dev/pandas/issues/50398/timeline,,,True,"{'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50398', 'html_url': 'https://github.com/pandas-dev/pandas/pull/50398', 'diff_url': 'https://github.com/pandas-dev/pandas/pull/50398.diff', 'patch_url': 'https://github.com/pandas-dev/pandas/pull/50398.patch', 'merged_at': None}"
21,https://api.github.com/repos/pandas-dev/pandas/issues/50396,https://api.github.com/repos/pandas-dev/pandas,https://api.github.com/repos/pandas-dev/pandas/issues/50396/labels{/name},https://api.github.com/repos/pandas-dev/pandas/issues/50396/comments,https://api.github.com/repos/pandas-dev/pandas/issues/50396/events,https://github.com/pandas-dev/pandas/pull/50396,1508154861,PR_kwDOAA0YD85GDmxd,50396,BUG/COMPAT: fix assert_* functions for nested arrays with latest numpy,"{'login': 'jorisvandenbossche', 'id': 1020496, 'node_id': 'MDQ6VXNlcjEwMjA0OTY=', 'avatar_url': 'https://avatars.githubusercontent.com/u/1020496?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/jorisvandenbossche', 'html_url': 'https://github.com/jorisvandenbossche', 'followers_url': 'https://api.github.com/users/jorisvandenbossche/followers', 'following_url': 'https://api.github.com/users/jorisvandenbossche/following{/other_user}', 'gists_url': 'https://api.github.com/users/jorisvandenbossche/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/jorisvandenbossche/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/jorisvandenbossche/subscriptions', 'organizations_url': 'https://api.github.com/users/jorisvandenbossche/orgs', 'repos_url': 'https://api.github.com/users/jorisvandenbossche/repos', 'events_url': 'https://api.github.com/users/jorisvandenbossche/events{/privacy}', 'received_events_url': 'https://api.github.com/users/jorisvandenbossche/received_events', 'type': 'User', 'site_admin': False}","[{'id': 127685, 'node_id': 'MDU6TGFiZWwxMjc2ODU=', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Testing', 'name': 'Testing', 'color': 'C4A000', 'default': False, 'description': 'pandas testing functions or related to the test suite'}, {'id': 76865106, 'node_id': 'MDU6TGFiZWw3Njg2NTEwNg==', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Compat', 'name': 'Compat', 'color': '0052cc', 'default': False, 'description': 'pandas objects compatability with Numpy or Python functions'}]",open,False,,[],"{'url': 'https://api.github.com/repos/pandas-dev/pandas/milestones/100', 'html_url': 'https://github.com/pandas-dev/pandas/milestone/100', 'labels_url': 'https://api.github.com/repos/pandas-dev/pandas/milestones/100/labels', 'id': 8654991, 'node_id': 'MI_kwDOAA0YD84AhBCP', 'number': 100, 'title': '1.5.3', 'description': 'on-merge: backport to 1.5.x', 'creator': {'login': 'datapythonista', 'id': 10058240, 'node_id': 'MDQ6VXNlcjEwMDU4MjQw', 'avatar_url': 'https://avatars.githubusercontent.com/u/10058240?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/datapythonista', 'html_url': 'https://github.com/datapythonista', 'followers_url': 'https://api.github.com/users/datapythonista/followers', 'following_url': 'https://api.github.com/users/datapythonista/following{/other_user}', 'gists_url': 'https://api.github.com/users/datapythonista/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/datapythonista/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/datapythonista/subscriptions', 'organizations_url': 'https://api.github.com/users/datapythonista/orgs', 'repos_url': 'https://api.github.com/users/datapythonista/repos', 'events_url': 'https://api.github.com/users/datapythonista/events{/privacy}', 'received_events_url': 'https://api.github.com/users/datapythonista/received_events', 'type': 'User', 'site_admin': False}, 'open_issues': 16, 'closed_issues': 55, 'state': 'open', 'created_at': '2022-11-15T04:46:08Z', 'updated_at': '2022-12-23T23:31:20Z', 'due_on': None, 'closed_at': None}",0,2022-12-22T16:17:39Z,2022-12-23T08:37:16Z,,MEMBER,,"This adds tests for `assert_almost_equal` (used by `assert_series_equal` et al in case of object dtype arrays) that all pass on pandas main with released numpy (except for two cases with dict of array), and fixes the implementation of `array_equivalent` (used by `assert_almost_equal`) to return False instead of raising an error if numpy cannot compare the arrays (numpy 1.25.dev starts to raise for this instead of returning False).

I also added a bunch of equivalent tests for `array_equivalent` itself, but those don't all pass (and added xfails). Trying to fix those as well might need more extensive changes, and I would prefer to do that separate from this PR (to keep this one possible to backport).

See https://github.com/pandas-dev/pandas/issues/50360#issuecomment-1362678681 for an illustration of the behaviour that changed in numpy 1.25.dev that causes this.

- [x] closes https://github.com/pandas-dev/pandas/issues/50360
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.
","{'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50396/reactions', 'total_count': 0, '+1': 0, '-1': 0, 'laugh': 0, 'hooray': 0, 'confused': 0, 'heart': 0, 'rocket': 0, 'eyes': 0}",https://api.github.com/repos/pandas-dev/pandas/issues/50396/timeline,,,False,"{'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50396', 'html_url': 'https://github.com/pandas-dev/pandas/pull/50396', 'diff_url': 'https://github.com/pandas-dev/pandas/pull/50396.diff', 'patch_url': 'https://github.com/pandas-dev/pandas/pull/50396.patch', 'merged_at': None}"
22,https://api.github.com/repos/pandas-dev/pandas/issues/50395,https://api.github.com/repos/pandas-dev/pandas,https://api.github.com/repos/pandas-dev/pandas/issues/50395/labels{/name},https://api.github.com/repos/pandas-dev/pandas/issues/50395/comments,https://api.github.com/repos/pandas-dev/pandas/issues/50395/events,https://github.com/pandas-dev/pandas/issues/50395,1508119987,I_kwDOAA0YD85Z5BWz,50395,ENH: Adding support for calamine as Excel reader engine,"{'login': 'abielr', 'id': 168236, 'node_id': 'MDQ6VXNlcjE2ODIzNg==', 'avatar_url': 'https://avatars.githubusercontent.com/u/168236?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/abielr', 'html_url': 'https://github.com/abielr', 'followers_url': 'https://api.github.com/users/abielr/followers', 'following_url': 'https://api.github.com/users/abielr/following{/other_user}', 'gists_url': 'https://api.github.com/users/abielr/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/abielr/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/abielr/subscriptions', 'organizations_url': 'https://api.github.com/users/abielr/orgs', 'repos_url': 'https://api.github.com/users/abielr/repos', 'events_url': 'https://api.github.com/users/abielr/events{/privacy}', 'received_events_url': 'https://api.github.com/users/abielr/received_events', 'type': 'User', 'site_admin': False}","[{'id': 76812, 'node_id': 'MDU6TGFiZWw3NjgxMg==', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Enhancement', 'name': 'Enhancement', 'color': '4E9A06', 'default': False, 'description': None}, {'id': 49254273, 'node_id': 'MDU6TGFiZWw0OTI1NDI3Mw==', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/IO%20Excel', 'name': 'IO Excel', 'color': 'bfe5bf', 'default': False, 'description': 'read_excel, to_excel'}]",open,False,,[],,1,2022-12-22T15:51:43Z,2022-12-22T16:37:47Z,,NONE,,"### Feature Type

- [X] Adding new functionality to pandas

- [ ] Changing existing functionality in pandas

- [ ] Removing existing functionality in pandas


### Problem Description

Reading Excel files in Pandas is considerably slower than in some alternative data frame tools, for example the `readxl` package in R can read Excel files much faster. The Rust `calamine` library can read Excel files much faster than other engines supported by Pandas, and there is an existing Python binding to it, [python-calamine](https://github.com/dimastbk/python-calamine). I would like to request that Pandas add official support for `calamine`, so that users can read an Excel file like:

`pd.read_excel(""test.xlsx"", engine=""calamine"")`

### Feature Description

The `python-calamine` package already implements code that enables the calamine engine in Pandas, see the examples using `pandas_monkeypatch()` at the bottom of their Github README. The code to enable this is [here](https://github.com/dimastbk/python-calamine/blob/master/python/python_calamine/pandas.py)

Although `python-calamine` already implements the necessary features to use the library with Pandas, I am unclear on how similar the behavior is between `calamine` and other engines that Pandas supports like `openpyxl`. I am hoping that by bringing `calamine` in as an officially supported engine that Pandas unit tests will confirm consistent behavior across `calamine` and other engines. 

### Alternative Solutions

None

### Additional Context

_No response_","{'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50395/reactions', 'total_count': 0, '+1': 0, '-1': 0, 'laugh': 0, 'hooray': 0, 'confused': 0, 'heart': 0, 'rocket': 0, 'eyes': 0}",https://api.github.com/repos/pandas-dev/pandas/issues/50395/timeline,,,,
23,https://api.github.com/repos/pandas-dev/pandas/issues/50392,https://api.github.com/repos/pandas-dev/pandas,https://api.github.com/repos/pandas-dev/pandas/issues/50392/labels{/name},https://api.github.com/repos/pandas-dev/pandas/issues/50392/comments,https://api.github.com/repos/pandas-dev/pandas/issues/50392/events,https://github.com/pandas-dev/pandas/issues/50392,1507777076,I_kwDOAA0YD85Z3to0,50392,CI unpin numpy,"{'login': 'MarcoGorelli', 'id': 33491632, 'node_id': 'MDQ6VXNlcjMzNDkxNjMy', 'avatar_url': 'https://avatars.githubusercontent.com/u/33491632?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/MarcoGorelli', 'html_url': 'https://github.com/MarcoGorelli', 'followers_url': 'https://api.github.com/users/MarcoGorelli/followers', 'following_url': 'https://api.github.com/users/MarcoGorelli/following{/other_user}', 'gists_url': 'https://api.github.com/users/MarcoGorelli/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/MarcoGorelli/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/MarcoGorelli/subscriptions', 'organizations_url': 'https://api.github.com/users/MarcoGorelli/orgs', 'repos_url': 'https://api.github.com/users/MarcoGorelli/repos', 'events_url': 'https://api.github.com/users/MarcoGorelli/events{/privacy}', 'received_events_url': 'https://api.github.com/users/MarcoGorelli/received_events', 'type': 'User', 'site_admin': False}","[{'id': 48070600, 'node_id': 'MDU6TGFiZWw0ODA3MDYwMA==', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/CI', 'name': 'CI', 'color': 'a2bca7', 'default': False, 'description': 'Continuous Integration'}]",open,False,,[],,0,2022-12-22T12:19:26Z,2022-12-22T12:19:26Z,,MEMBER,,"numpy 1.24.0 can't be used with numba. once https://github.com/numba/numba/pull/8620 is merged, and this is fixed in numba, we can unpin numpy in CI","{'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50392/reactions', 'total_count': 0, '+1': 0, '-1': 0, 'laugh': 0, 'hooray': 0, 'confused': 0, 'heart': 0, 'rocket': 0, 'eyes': 0}",https://api.github.com/repos/pandas-dev/pandas/issues/50392/timeline,,,,
24,https://api.github.com/repos/pandas-dev/pandas/issues/50390,https://api.github.com/repos/pandas-dev/pandas,https://api.github.com/repos/pandas-dev/pandas/issues/50390/labels{/name},https://api.github.com/repos/pandas-dev/pandas/issues/50390/comments,https://api.github.com/repos/pandas-dev/pandas/issues/50390/events,https://github.com/pandas-dev/pandas/issues/50390,1507564558,I_kwDOAA0YD85Z25wO,50390,ENH: Rolling with window=offset and step=offset,"{'login': 'harrylojames', 'id': 51630653, 'node_id': 'MDQ6VXNlcjUxNjMwNjUz', 'avatar_url': 'https://avatars.githubusercontent.com/u/51630653?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/harrylojames', 'html_url': 'https://github.com/harrylojames', 'followers_url': 'https://api.github.com/users/harrylojames/followers', 'following_url': 'https://api.github.com/users/harrylojames/following{/other_user}', 'gists_url': 'https://api.github.com/users/harrylojames/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/harrylojames/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/harrylojames/subscriptions', 'organizations_url': 'https://api.github.com/users/harrylojames/orgs', 'repos_url': 'https://api.github.com/users/harrylojames/repos', 'events_url': 'https://api.github.com/users/harrylojames/events{/privacy}', 'received_events_url': 'https://api.github.com/users/harrylojames/received_events', 'type': 'User', 'site_admin': False}","[{'id': 76812, 'node_id': 'MDU6TGFiZWw3NjgxMg==', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Enhancement', 'name': 'Enhancement', 'color': '4E9A06', 'default': False, 'description': None}, {'id': 1954720290, 'node_id': 'MDU6TGFiZWwxOTU0NzIwMjkw', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Needs%20Triage', 'name': 'Needs Triage', 'color': '0052cc', 'default': False, 'description': 'Issue that has not been reviewed by a pandas team member'}]",open,False,,[],,0,2022-12-22T09:50:05Z,2022-12-22T09:50:05Z,,NONE,,"### Feature Type

- [X] Adding new functionality to pandas

- [ ] Changing existing functionality in pandas

- [ ] Removing existing functionality in pandas


### Problem Description

Would be really great to be able to write df.rolling(window=""7d"", step=""1d)

Currently this returns NotImplementedError: step is not supported with frequency windows

### Feature Description

.

### Alternative Solutions

.

### Additional Context

_No response_","{'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50390/reactions', 'total_count': 0, '+1': 0, '-1': 0, 'laugh': 0, 'hooray': 0, 'confused': 0, 'heart': 0, 'rocket': 0, 'eyes': 0}",https://api.github.com/repos/pandas-dev/pandas/issues/50390/timeline,,,,
25,https://api.github.com/repos/pandas-dev/pandas/issues/50388,https://api.github.com/repos/pandas-dev/pandas,https://api.github.com/repos/pandas-dev/pandas/issues/50388/labels{/name},https://api.github.com/repos/pandas-dev/pandas/issues/50388/comments,https://api.github.com/repos/pandas-dev/pandas/issues/50388/events,https://github.com/pandas-dev/pandas/pull/50388,1507017355,PR_kwDOAA0YD85F_tAP,50388,WEB: Add backoff + retry for Github API calls,"{'login': 'mroeschke', 'id': 10647082, 'node_id': 'MDQ6VXNlcjEwNjQ3MDgy', 'avatar_url': 'https://avatars.githubusercontent.com/u/10647082?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/mroeschke', 'html_url': 'https://github.com/mroeschke', 'followers_url': 'https://api.github.com/users/mroeschke/followers', 'following_url': 'https://api.github.com/users/mroeschke/following{/other_user}', 'gists_url': 'https://api.github.com/users/mroeschke/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/mroeschke/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/mroeschke/subscriptions', 'organizations_url': 'https://api.github.com/users/mroeschke/orgs', 'repos_url': 'https://api.github.com/users/mroeschke/repos', 'events_url': 'https://api.github.com/users/mroeschke/events{/privacy}', 'received_events_url': 'https://api.github.com/users/mroeschke/received_events', 'type': 'User', 'site_admin': False}","[{'id': 48070600, 'node_id': 'MDU6TGFiZWw0ODA3MDYwMA==', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/CI', 'name': 'CI', 'color': 'a2bca7', 'default': False, 'description': 'Continuous Integration'}, {'id': 1508144531, 'node_id': 'MDU6TGFiZWwxNTA4MTQ0NTMx', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Web', 'name': 'Web', 'color': '23ef97', 'default': False, 'description': 'pandas website'}]",open,False,,[],,1,2022-12-21T23:02:48Z,2022-12-22T17:22:00Z,,MEMBER,,"Hopefully to help the doc and web build sometimes failing because of hitting the rate limit of the GIthub API

```
Traceback (most recent call last):
  File ""web/pandas_web.py"", line 403, in <module>
    main(args.source_path, args.target_path, args.base_url, args.ignore_io_errors)
  File ""web/pandas_web.py"", line 349, in main
    context = get_context(config_fname, ignore_io_errors, base_url=base_url)
  File ""web/pandas_web.py"", line 305, in get_context
    context = preprocessor(context)
  File ""/home/runner/work/pandas/pandas/web/pandas_web.py"", line 172, in maintainers_add_info
    resp.raise_for_status()
  File ""/home/runner/micromamba/envs/test/lib/python3.8/site-packages/requests/models.py"", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 403 Client Error: rate limit exceeded for url: https://api.github.com/users/shoyer

```","{'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50388/reactions', 'total_count': 0, '+1': 0, '-1': 0, 'laugh': 0, 'hooray': 0, 'confused': 0, 'heart': 0, 'rocket': 0, 'eyes': 0}",https://api.github.com/repos/pandas-dev/pandas/issues/50388/timeline,,,False,"{'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50388', 'html_url': 'https://github.com/pandas-dev/pandas/pull/50388', 'diff_url': 'https://github.com/pandas-dev/pandas/pull/50388.diff', 'patch_url': 'https://github.com/pandas-dev/pandas/pull/50388.patch', 'merged_at': None}"
26,https://api.github.com/repos/pandas-dev/pandas/issues/50386,https://api.github.com/repos/pandas-dev/pandas,https://api.github.com/repos/pandas-dev/pandas/issues/50386/labels{/name},https://api.github.com/repos/pandas-dev/pandas/issues/50386/comments,https://api.github.com/repos/pandas-dev/pandas/issues/50386/events,https://github.com/pandas-dev/pandas/pull/50386,1506936544,PR_kwDOAA0YD85F_b5P,50386,CI: Test pandas warnings as error on some builds,"{'login': 'mroeschke', 'id': 10647082, 'node_id': 'MDQ6VXNlcjEwNjQ3MDgy', 'avatar_url': 'https://avatars.githubusercontent.com/u/10647082?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/mroeschke', 'html_url': 'https://github.com/mroeschke', 'followers_url': 'https://api.github.com/users/mroeschke/followers', 'following_url': 'https://api.github.com/users/mroeschke/following{/other_user}', 'gists_url': 'https://api.github.com/users/mroeschke/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/mroeschke/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/mroeschke/subscriptions', 'organizations_url': 'https://api.github.com/users/mroeschke/orgs', 'repos_url': 'https://api.github.com/users/mroeschke/repos', 'events_url': 'https://api.github.com/users/mroeschke/events{/privacy}', 'received_events_url': 'https://api.github.com/users/mroeschke/received_events', 'type': 'User', 'site_admin': False}",[],open,False,,[],,1,2022-12-21T21:32:24Z,2022-12-23T23:36:35Z,,MEMBER,,Locally I was validating the recent addition of `-W error:::pandas` and had to change it to `-W error:::$pandas` so validating if the CI will raise as well,"{'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50386/reactions', 'total_count': 0, '+1': 0, '-1': 0, 'laugh': 0, 'hooray': 0, 'confused': 0, 'heart': 0, 'rocket': 0, 'eyes': 0}",https://api.github.com/repos/pandas-dev/pandas/issues/50386/timeline,,,True,"{'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50386', 'html_url': 'https://github.com/pandas-dev/pandas/pull/50386', 'diff_url': 'https://github.com/pandas-dev/pandas/pull/50386.diff', 'patch_url': 'https://github.com/pandas-dev/pandas/pull/50386.patch', 'merged_at': None}"
27,https://api.github.com/repos/pandas-dev/pandas/issues/50383,https://api.github.com/repos/pandas-dev/pandas,https://api.github.com/repos/pandas-dev/pandas/issues/50383/labels{/name},https://api.github.com/repos/pandas-dev/pandas/issues/50383/comments,https://api.github.com/repos/pandas-dev/pandas/issues/50383/events,https://github.com/pandas-dev/pandas/issues/50383,1506751371,I_kwDOAA0YD85ZzzOL,50383,BUG: Inconsistency with grouping columns in agg,"{'login': 'rhshadrach', 'id': 45562402, 'node_id': 'MDQ6VXNlcjQ1NTYyNDAy', 'avatar_url': 'https://avatars.githubusercontent.com/u/45562402?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/rhshadrach', 'html_url': 'https://github.com/rhshadrach', 'followers_url': 'https://api.github.com/users/rhshadrach/followers', 'following_url': 'https://api.github.com/users/rhshadrach/following{/other_user}', 'gists_url': 'https://api.github.com/users/rhshadrach/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/rhshadrach/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/rhshadrach/subscriptions', 'organizations_url': 'https://api.github.com/users/rhshadrach/orgs', 'repos_url': 'https://api.github.com/users/rhshadrach/repos', 'events_url': 'https://api.github.com/users/rhshadrach/events{/privacy}', 'received_events_url': 'https://api.github.com/users/rhshadrach/received_events', 'type': 'User', 'site_admin': False}","[{'id': 76811, 'node_id': 'MDU6TGFiZWw3NjgxMQ==', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Bug', 'name': 'Bug', 'color': 'e10c02', 'default': False, 'description': None}, {'id': 233160, 'node_id': 'MDU6TGFiZWwyMzMxNjA=', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Groupby', 'name': 'Groupby', 'color': '729FCF', 'default': False, 'description': None}]",open,False,,[],,0,2022-12-21T18:39:05Z,2022-12-21T18:39:05Z,,MEMBER,,"Related to #46944

Users can currently request the grouping column be part of the computation for various ops by including them as part of a `__getitem__`. But `agg` will still exclude these columns.

```
df = pd.DataFrame({'a': [1, 1, 2], 'b': 3, 'c': 4, 'd': 5})
gb = df.groupby(['a', 'b'])[['a', 'c']]

result = gb.sum()
print(result)
#      a  c
# a b      
# 1 3  2  8
# 2 3  2  4

result2 = gb.agg(lambda x: x.sum())
print(result2)
#      c
# a b   
# 1 3  8
# 2 3  4
```

I would expect `__getitem__` to only subset columns for groupby rather than being able to add additional (grouping) columns.","{'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50383/reactions', 'total_count': 0, '+1': 0, '-1': 0, 'laugh': 0, 'hooray': 0, 'confused': 0, 'heart': 0, 'rocket': 0, 'eyes': 0}",https://api.github.com/repos/pandas-dev/pandas/issues/50383/timeline,,,,
28,https://api.github.com/repos/pandas-dev/pandas/issues/50382,https://api.github.com/repos/pandas-dev/pandas,https://api.github.com/repos/pandas-dev/pandas/issues/50382/labels{/name},https://api.github.com/repos/pandas-dev/pandas/issues/50382/comments,https://api.github.com/repos/pandas-dev/pandas/issues/50382/events,https://github.com/pandas-dev/pandas/pull/50382,1506628703,PR_kwDOAA0YD85F-YyA,50382,ENH: Add std to masked interface,"{'login': 'phofl', 'id': 61934744, 'node_id': 'MDQ6VXNlcjYxOTM0NzQ0', 'avatar_url': 'https://avatars.githubusercontent.com/u/61934744?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/phofl', 'html_url': 'https://github.com/phofl', 'followers_url': 'https://api.github.com/users/phofl/followers', 'following_url': 'https://api.github.com/users/phofl/following{/other_user}', 'gists_url': 'https://api.github.com/users/phofl/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/phofl/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/phofl/subscriptions', 'organizations_url': 'https://api.github.com/users/phofl/orgs', 'repos_url': 'https://api.github.com/users/phofl/repos', 'events_url': 'https://api.github.com/users/phofl/events{/privacy}', 'received_events_url': 'https://api.github.com/users/phofl/received_events', 'type': 'User', 'site_admin': False}","[{'id': 1817503692, 'node_id': 'MDU6TGFiZWwxODE3NTAzNjky', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/NA%20-%20MaskedArrays', 'name': 'NA - MaskedArrays', 'color': '8cc645', 'default': False, 'description': 'Related to pd.NA and nullable extension arrays'}, {'id': 2365504383, 'node_id': 'MDU6TGFiZWwyMzY1NTA0Mzgz', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Reduction%20Operations', 'name': 'Reduction Operations', 'color': '547c03', 'default': False, 'description': 'sum, mean, min, max, etc.'}]",open,False,,[],,0,2022-12-21T16:59:52Z,2022-12-22T12:00:01Z,,MEMBER,,"- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

The warnings were already shown before I changed the code paths, but not caught
","{'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50382/reactions', 'total_count': 0, '+1': 0, '-1': 0, 'laugh': 0, 'hooray': 0, 'confused': 0, 'heart': 0, 'rocket': 0, 'eyes': 0}",https://api.github.com/repos/pandas-dev/pandas/issues/50382/timeline,,,False,"{'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50382', 'html_url': 'https://github.com/pandas-dev/pandas/pull/50382', 'diff_url': 'https://github.com/pandas-dev/pandas/pull/50382.diff', 'patch_url': 'https://github.com/pandas-dev/pandas/pull/50382.patch', 'merged_at': None}"
29,https://api.github.com/repos/pandas-dev/pandas/issues/50380,https://api.github.com/repos/pandas-dev/pandas,https://api.github.com/repos/pandas-dev/pandas/issues/50380/labels{/name},https://api.github.com/repos/pandas-dev/pandas/issues/50380/comments,https://api.github.com/repos/pandas-dev/pandas/issues/50380/events,https://github.com/pandas-dev/pandas/issues/50380,1506568111,I_kwDOAA0YD85ZzGev,50380,TST 56 tests in pandas/tests/test_generic.py aren't being tested,"{'login': 'MarcoGorelli', 'id': 33491632, 'node_id': 'MDQ6VXNlcjMzNDkxNjMy', 'avatar_url': 'https://avatars.githubusercontent.com/u/33491632?v=4', 'gravatar_id': '', 'url': 'https://api.github.com/users/MarcoGorelli', 'html_url': 'https://github.com/MarcoGorelli', 'followers_url': 'https://api.github.com/users/MarcoGorelli/followers', 'following_url': 'https://api.github.com/users/MarcoGorelli/following{/other_user}', 'gists_url': 'https://api.github.com/users/MarcoGorelli/gists{/gist_id}', 'starred_url': 'https://api.github.com/users/MarcoGorelli/starred{/owner}{/repo}', 'subscriptions_url': 'https://api.github.com/users/MarcoGorelli/subscriptions', 'organizations_url': 'https://api.github.com/users/MarcoGorelli/orgs', 'repos_url': 'https://api.github.com/users/MarcoGorelli/repos', 'events_url': 'https://api.github.com/users/MarcoGorelli/events{/privacy}', 'received_events_url': 'https://api.github.com/users/MarcoGorelli/received_events', 'type': 'User', 'site_admin': False}","[{'id': 986278782, 'node_id': 'MDU6TGFiZWw5ODYyNzg3ODI=', 'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Needs%20Tests', 'name': 'Needs Tests', 'color': 'cdea3c', 'default': False, 'description': 'Unit test(s) needed to prevent regressions'}]",open,False,,[],,0,2022-12-21T16:15:50Z,2022-12-21T16:18:43Z,,MEMBER,,"All these tests aren't actually being run:

https://github.com/pandas-dev/pandas/blob/2ffb74814fb7ba800ed4ec6bc1935ccb78f9cb65/pandas/tests/generic/test_generic.py#L53-L306

that's because the class name is `Generic`, rather than `TestGeneric`

They're all uncovered in the coverage report: https://app.codecov.io/gh/pandas-dev/pandas/blob/main/pandas/tests/generic/test_generic.py

Similar to https://github.com/pandas-dev/pandas/issues/50378, the class needs renaming, and the tests fixing up","{'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50380/reactions', 'total_count': 0, '+1': 0, '-1': 0, 'laugh': 0, 'hooray': 0, 'confused': 0, 'heart': 0, 'rocket': 0, 'eyes': 0}",https://api.github.com/repos/pandas-dev/pandas/issues/50380/timeline,,,,
